{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27acbbe6-3ec2-4ea2-824b-e4db27a2cbd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "## ATAC-SEQ DATA MAPPING\n",
    "# Get data (example)\n",
    "data_dir=\"/staging/leuven/stg_00002/lcb/ngs_runs/NextSeq2000_20230321/Demultiplexed\"\n",
    "# Prepare\n",
    "module load FastQC\n",
    "module load ea-utils/20170620-foss-2018a\n",
    "module load deepTools/3.5.0-foss-2018a-Python-3.7.4\n",
    "module load Bowtie2/2.3.5.1-foss-2018a\n",
    "module load SAMtools/1.10-foss-2018a\n",
    "module load MACS2/2.1.2.1-foss-2018a-Python-2.7.16\n",
    "dir='/staging/leuven/stg_00002/lcb/cbravo/Liver/Hepa16/ATAC'\n",
    "MAXCORES=7\n",
    "# Create directories\n",
    "mkdir ${dir}\n",
    "mkdir ${dir}/clean_fastq/\n",
    "mkdir ${dir}/fastqc/\n",
    "mkdir ${dir}/bam/\n",
    "mkdir ${dir}/MACS/\n",
    "#File\n",
    "filename='MLV__babe04__OmniATAC_Hepa1-6_S3'\n",
    "# Clean reads\n",
    "fastq-mcf \\\n",
    "/staging/leuven/stg_00002/lcb/kdavie/Software/bin/ea-utils.1.1.2-686/fastqmcf.adapters \\\n",
    "${data_dir}/${filename}_R1_001.fastq.gz \\\n",
    "${data_dir}/${filename}_R2_001.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_1.clean.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_2.clean.fastq.gz\n",
    "# QC\n",
    "for file in ${dir}/clean_fastq/${filename}*\n",
    "do\n",
    "    fastqc -t ${MAXCORES} ${file} -o ${dir}/fastqc/\n",
    "done\n",
    "# Mapping\n",
    "bowtie2 -p 5 -x /staging/leuven/res_00001/genomes/mus_musculus/mm10_iGenomes/indexes/Bowtie2/genome -1 ${dir}/clean_fastq/${filename}_1.clean.fastq.gz -2 ${dir}/clean_fastq/${filename}_2.clean.fastq.gz -S ${dir}/bam/${filename}.sam\n",
    "\n",
    "samtools view -hu -q 4 ${dir}/bam/${filename}.sam | samtools sort \\\n",
    "        -@ ${MAXCORES} \\\n",
    "        -O bam \\\n",
    "        -T ${dir}/bam/tmp.sam \\\n",
    "        -o ${dir}/bam/${filename}_q4_sorted.bam \n",
    "samtools index ${dir}/bam/${filename}_q4_sorted.bam\n",
    "#rm ${dir}/bam/${filename}.sam\n",
    "# Peak calling\n",
    "macs2 callpeak --treatment ${dir}/bam/${filename}_q4_sorted.bam  --name ${filename} --outdir ${dir}/MACS/ --format BAMPE --gsize mm --qvalue 0.05 --nomodel --shift 73 --extsize 146 --keep-dup all --call-summits --nolambda\n",
    "bamCoverage -b ${dir}/bam/${filename}_q4_sorted.bam -o ${dir}/bam/${filename}_q4_sorted.bw --normalizeUsing CPM --binSize 1 -p 10\n",
    "echo 'Done!' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e68737c3-1d93-490c-8e76-e5601347d0f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "## ATAC-SEQ DATA MAPPING\n",
    "# Get data (example)\n",
    "data_dir=\"/staging/leuven/stg_00002/lcb/ngs_runs/NextSeq2000_20230321/Demultiplexed\"\n",
    "# Prepare\n",
    "module load FastQC\n",
    "module load ea-utils/20170620-foss-2018a\n",
    "module load deepTools/3.5.0-foss-2018a-Python-3.7.4\n",
    "module load Bowtie2/2.3.5.1-foss-2018a\n",
    "module load SAMtools/1.10-foss-2018a\n",
    "module load MACS2/2.1.2.1-foss-2018a-Python-2.7.16\n",
    "dir='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3'\n",
    "MAXCORES=7\n",
    "# Create directories\n",
    "mkdir ${dir}\n",
    "mkdir ${dir}/clean_fastq/\n",
    "mkdir ${dir}/fastqc/\n",
    "mkdir ${dir}/bam/\n",
    "mkdir ${dir}/MACS/\n",
    "#File\n",
    "filename='MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29'\n",
    "# Clean reads\n",
    "fastq-mcf \\\n",
    "/staging/leuven/stg_00002/lcb/kdavie/Software/bin/ea-utils.1.1.2-686/fastqmcf.adapters \\\n",
    "${data_dir}/${filename}_R1_001.fastq.gz \\\n",
    "${data_dir}/${filename}_R2_001.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_1.clean.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_2.clean.fastq.gz\n",
    "# QC\n",
    "for file in ${dir}/clean_fastq/${filename}*\n",
    "do\n",
    "    fastqc -t ${MAXCORES} ${file} -o ${dir}/fastqc/\n",
    "done\n",
    "# Mapping\n",
    "bowtie2 -p 5 -x /staging/leuven/res_00001/genomes/mus_musculus/mm10_iGenomes/indexes/Bowtie2/genome -1 ${dir}/clean_fastq/${filename}_1.clean.fastq.gz -2 ${dir}/clean_fastq/${filename}_2.clean.fastq.gz -S ${dir}/bam/${filename}.sam\n",
    "\n",
    "samtools view -hu -q 4 ${dir}/bam/${filename}.sam | samtools sort \\\n",
    "        -@ ${MAXCORES} \\\n",
    "        -O bam \\\n",
    "        -T ${dir}/bam/tmp.sam \\\n",
    "        -o ${dir}/bam/${filename}_q4_sorted.bam \n",
    "samtools index ${dir}/bam/${filename}_q4_sorted.bam\n",
    "#rm ${dir}/bam/${filename}.sam\n",
    "# Peak calling\n",
    "macs2 callpeak --treatment ${dir}/bam/${filename}_q4_sorted.bam  --name ${filename} --outdir ${dir}/MACS/ --format BAMPE --gsize mm --qvalue 0.05 --nomodel --shift 73 --extsize 146 --keep-dup all --call-summits --nolambda\n",
    "\n",
    "bamCoverage -b ${dir}/bam/${filename}_q4_sorted.bam -o ${dir}/bam/${filename}_q4_sorted.bw --normalizeUsing CPM --binSize 1 -p 10\n",
    "echo 'Done!' \n",
    "\n",
    "\n",
    "module load MACS2/2.1.2.1-foss-2018a-Python-2.7.16\n",
    "dir='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3'\n",
    "mkdir ${dir}/MACS0.01/ \n",
    "filename='MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29'\n",
    "macs2 callpeak -c ${dir}/bam/MLV__232339__ChIPseq_mouse_hepatocytes_input_S31_q4_sorted.bam -t ${dir}/bam/${filename}_q4_sorted.bam --name ${filename}_VS_CONTROL --outdir ${dir}/MACS0.01/ --format BAMPE --gsize mm --qvalue 0.01 --call-summits\n",
    "#dir='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3'\n",
    "\n",
    "module load MACS2/2.1.2.1-foss-2018a-Python-2.7.16\n",
    "dir='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3'\n",
    "mkdir ${dir}/MACS0.05/ \n",
    "filename='MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29'\n",
    "macs2 callpeak -c ${dir}/bam/MLV__232339__ChIPseq_mouse_hepatocytes_input_S31_q4_sorted.bam -t ${dir}/bam/${filename}_q4_sorted.bam --name ${filename}_VS_CONTROL --outdir ${dir}/MACS0.05/ --format BAMPE --gsize mm --qvalue 0.05 --call-summits\n",
    "#mkdir ${dir}/MACS0.001/ \n",
    "#filename='MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29'\n",
    "#macs2 callpeak --treatment ${dir}/bam/${filename}_q4_sorted.bam --control ${dir}/bam/MLV__232339__ChIPseq_mouse_hepatocytes_input_S31_q4_sorted.bam  --name ${filename}_VS_CONTROL --outdir ${dir}/MACS0.01/ --format BAMPE --gsize mm --qvalue 0.001 --nomodel  --keep-dup all --call-summits --nolambda\n",
    "\n",
    "#filename='MLV__141ec0__ChIPseq_mouse_hepatocytes_anti-TBX3_Santa_Cruz_mouse_S30'\n",
    "#macs2 callpeak --treatment ${dir}/bam/${filename}_q4_sorted.bam --control ${dir}/bam/MLV__232339__ChIPseq_mouse_hepatocytes_input_S31_q4_sorted.bam  --name ${filename}_VS_CONTROL --outdir ${dir}/MACS/ --format BAM --gsize mm --qvalue 0.05 --nomodel  --keep-dup all --call-summits --nolambda\n",
    "\n",
    "#File\n",
    "filename='MLV__141ec0__ChIPseq_mouse_hepatocytes_anti-TBX3_Santa_Cruz_mouse_S30'\n",
    "# Clean reads\n",
    "fastq-mcf \\\n",
    "/staging/leuven/stg_00002/lcb/kdavie/Software/bin/ea-utils.1.1.2-686/fastqmcf.adapters \\\n",
    "${data_dir}/${filename}_R1_001.fastq.gz \\\n",
    "${data_dir}/${filename}_R2_001.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_1.clean.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_2.clean.fastq.gz\n",
    "# QC\n",
    "for file in ${dir}/clean_fastq/${filename}*\n",
    "do\n",
    "    fastqc -t ${MAXCORES} ${file} -o ${dir}/fastqc/\n",
    "done\n",
    "# Mapping\n",
    "bowtie2 -p 5 -x /staging/leuven/res_00001/genomes/mus_musculus/mm10_iGenomes/indexes/Bowtie2/genome -1 ${dir}/clean_fastq/${filename}_1.clean.fastq.gz -2 ${dir}/clean_fastq/${filename}_2.clean.fastq.gz -S ${dir}/bam/${filename}.sam\n",
    "\n",
    "samtools view -hu -q 4 ${dir}/bam/${filename}.sam | samtools sort \\\n",
    "        -@ ${MAXCORES} \\\n",
    "        -O bam \\\n",
    "        -T ${dir}/bam/tmp.sam \\\n",
    "        -o ${dir}/bam/${filename}_q4_sorted.bam \n",
    "samtools index ${dir}/bam/${filename}_q4_sorted.bam\n",
    "#rm ${dir}/bam/${filename}.sam\n",
    "# Peak calling\n",
    "macs2 callpeak --treatment ${dir}/bam/${filename}_q4_sorted.bam  --name ${filename} --outdir ${dir}/MACS/ --format BAMPE --gsize mm --qvalue 0.05 --nomodel --shift 73 --extsize 146 --keep-dup all --call-summits --nolambda\n",
    "bamCoverage -b ${dir}/bam/${filename}_q4_sorted.bam -o ${dir}/bam/${filename}_q4_sorted.bw --normalizeUsing CPM --binSize 1 -p 10\n",
    "echo 'Done!' \n",
    "\n",
    "filename='MLV__232339__ChIPseq_mouse_hepatocytes_input_S31'\n",
    "# Clean reads\n",
    "fastq-mcf \\\n",
    "/staging/leuven/stg_00002/lcb/kdavie/Software/bin/ea-utils.1.1.2-686/fastqmcf.adapters \\\n",
    "${data_dir}/${filename}_R1_001.fastq.gz \\\n",
    "${data_dir}/${filename}_R2_001.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_1.clean.fastq.gz \\\n",
    "-o ${dir}/clean_fastq/${filename}_2.clean.fastq.gz\n",
    "# QC\n",
    "for file in ${dir}/clean_fastq/${filename}*\n",
    "do\n",
    "    fastqc -t ${MAXCORES} ${file} -o ${dir}/fastqc/\n",
    "done\n",
    "# Mapping\n",
    "bowtie2 -p 5 -x /staging/leuven/res_00001/genomes/mus_musculus/mm10_iGenomes/indexes/Bowtie2/genome -1 ${dir}/clean_fastq/${filename}_1.clean.fastq.gz -2 ${dir}/clean_fastq/${filename}_2.clean.fastq.gz -S ${dir}/bam/${filename}.sam\n",
    "\n",
    "samtools view -hu -q 4 ${dir}/bam/${filename}.sam | samtools sort \\\n",
    "        -@ ${MAXCORES} \\\n",
    "        -O bam \\\n",
    "        -T ${dir}/bam/tmp.sam \\\n",
    "        -o ${dir}/bam/${filename}_q4_sorted.bam \n",
    "samtools index ${dir}/bam/${filename}_q4_sorted.bam\n",
    "#rm ${dir}/bam/${filename}.sam\n",
    "# Peak calling\n",
    "macs2 callpeak --treatment ${dir}/bam/${filename}_q4_sorted.bam  --name ${filename} --outdir ${dir}/MACS/ --format BAMPE --gsize mm --qvalue 0.05 --nomodel --shift 73 --extsize 146 --call-summits --nolambda\n",
    "bamCoverage -b ${dir}/bam/${filename}_q4_sorted.bam -o ${dir}/bam/${filename}_q4_sorted.bw --normalizeUsing CPM --binSize 1 -p 10\n",
    "echo 'Done!' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c664039d-9834-4dcd-8cda-cf4b3defe912",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pyranges as pr\n",
    "import os\n",
    "path_to_region_sets = '/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/'\n",
    "region_sets_files = ['MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_VS_CONTROL_peaks.narrowPeak']\n",
    "region_sets = {'All_peaks' :pr.read_bed(os.path.join(path_to_region_sets, x)) for x in region_sets_files}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "54a03534-693a-4be9-aa29-a405da31f33d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[+--------------+-----------+-----------+-------+\n",
       " | Chromosome   | Start     | End       | +7    |\n",
       " | (category)   | (int32)   | (int32)   | ...   |\n",
       " |--------------+-----------+-----------+-------|\n",
       " | chr1         | 3514921   | 3515272   | ...   |\n",
       " | chr1         | 4513848   | 4514360   | ...   |\n",
       " | chr1         | 4571674   | 4571951   | ...   |\n",
       " | chr1         | 4598563   | 4598869   | ...   |\n",
       " | ...          | ...       | ...       | ...   |\n",
       " | chrY         | 1010362   | 1010612   | ...   |\n",
       " | chrY         | 9438185   | 9438665   | ...   |\n",
       " | chrY         | 9717927   | 9718141   | ...   |\n",
       " | chrY         | 50992937  | 50993222  | ...   |\n",
       " +--------------+-----------+-----------+-------+\n",
       " Unstranded PyRanges object has 23,951 rows and 10 columns from 21 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.\n",
       " Considered unstranded due to these Strand values: '.'\n",
       " 7 hidden columns: Name, Score, Strand, ThickStart, ... (+ 3 more.),\n",
       " +--------------+-----------+-----------+-------+\n",
       " | Chromosome   | Start     | End       | +7    |\n",
       " | (category)   | (int32)   | (int32)   | ...   |\n",
       " |--------------+-----------+-----------+-------|\n",
       " | chr1         | 67116156  | 67117327  | ...   |\n",
       " | chr1         | 162963456 | 162964057 | ...   |\n",
       " | chr1         | 66862147  | 66863109  | ...   |\n",
       " | chr1         | 88278361  | 88279628  | ...   |\n",
       " | ...          | ...       | ...       | ...   |\n",
       " | chrX         | 36599967  | 36600201  | ...   |\n",
       " | chrX         | 60348871  | 60349108  | ...   |\n",
       " | chrX         | 13041749  | 13042072  | ...   |\n",
       " | chrX         | 160412593 | 160413219 | ...   |\n",
       " | chrY         | 9717927   | 9718141   | ...   |\n",
       " | chrY         | 9438185   | 9438665   | ...   |\n",
       " +--------------+-----------+-----------+-------+\n",
       " Unstranded PyRanges object has 20,000 rows and 10 columns from 21 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.\n",
       " Considered unstranded due to these Strand values: '.'\n",
       " 7 hidden columns: Name, Score, Strand, ThickStart, ... (+ 3 more.),\n",
       " +--------------+-----------+-----------+-------+\n",
       " | Chromosome   | Start     | End       | +7    |\n",
       " | (category)   | (int32)   | (int32)   | ...   |\n",
       " |--------------+-----------+-----------+-------|\n",
       " | chr1         | 67116156  | 67117327  | ...   |\n",
       " | chr1         | 162963456 | 162964057 | ...   |\n",
       " | chr1         | 66862147  | 66863109  | ...   |\n",
       " | chr1         | 88278361  | 88279628  | ...   |\n",
       " | ...          | ...       | ...       | ...   |\n",
       " | chrX         | 57392245  | 57392982  | ...   |\n",
       " | chrX         | 36794405  | 36795011  | ...   |\n",
       " | chrX         | 85292857  | 85293089  | ...   |\n",
       " | chrX         | 145455596 | 145455908 | ...   |\n",
       " | chrY         | 9438185   | 9438665   | ...   |\n",
       " +--------------+-----------+-----------+-------+\n",
       " Unstranded PyRanges object has 10,000 rows and 10 columns from 21 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.\n",
       " Considered unstranded due to these Strand values: '.'\n",
       " 7 hidden columns: Name, Score, Strand, ThickStart, ... (+ 3 more.),\n",
       " +--------------+-----------+-----------+-------+\n",
       " | Chromosome   | Start     | End       | +7    |\n",
       " | (category)   | (int32)   | (int32)   | ...   |\n",
       " |--------------+-----------+-----------+-------|\n",
       " | chr1         | 67116156  | 67117327  | ...   |\n",
       " | chr1         | 162963456 | 162964057 | ...   |\n",
       " | chr1         | 66862147  | 66863109  | ...   |\n",
       " | chr1         | 88278361  | 88279628  | ...   |\n",
       " | ...          | ...       | ...       | ...   |\n",
       " | chrX         | 42044632  | 42045322  | ...   |\n",
       " | chrX         | 7840852   | 7841183   | ...   |\n",
       " | chrX         | 12761745  | 12762406  | ...   |\n",
       " | chrX         | 139684979 | 139685638 | ...   |\n",
       " +--------------+-----------+-----------+-------+\n",
       " Unstranded PyRanges object has 5,000 rows and 10 columns from 20 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.\n",
       " Considered unstranded due to these Strand values: '.'\n",
       " 7 hidden columns: Name, Score, Strand, ThickStart, ... (+ 3 more.),\n",
       " +--------------+-----------+-----------+-------+\n",
       " | Chromosome   | Start     | End       | +7    |\n",
       " | (category)   | (int32)   | (int32)   | ...   |\n",
       " |--------------+-----------+-----------+-------|\n",
       " | chr1         | 67116156  | 67117327  | ...   |\n",
       " | chr1         | 162963456 | 162964057 | ...   |\n",
       " | chr1         | 66862147  | 66863109  | ...   |\n",
       " | chr1         | 88278361  | 88279628  | ...   |\n",
       " | ...          | ...       | ...       | ...   |\n",
       " | chr19        | 37442375  | 37443106  | ...   |\n",
       " | chr19        | 40222736  | 40223242  | ...   |\n",
       " | chr19        | 42510147  | 42510804  | ...   |\n",
       " | chr19        | 57196795  | 57198052  | ...   |\n",
       " | chrX         | 140456167 | 140456817 | ...   |\n",
       " | chrX         | 36126211  | 36126785  | ...   |\n",
       " | chrX         | 47992489  | 47992859  | ...   |\n",
       " +--------------+-----------+-----------+-------+\n",
       " Unstranded PyRanges object has 1,000 rows and 10 columns from 20 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.\n",
       " Considered unstranded due to these Strand values: '.'\n",
       " 7 hidden columns: Name, Score, Strand, ThickStart, ... (+ 3 more.),\n",
       " +--------------+-----------+-----------+-------+\n",
       " | Chromosome   | Start     | End       | +7    |\n",
       " | (category)   | (int32)   | (int32)   | ...   |\n",
       " |--------------+-----------+-----------+-------|\n",
       " | chr1         | 8876494   | 8877004   | ...   |\n",
       " | chr1         | 33734212  | 33734523  | ...   |\n",
       " | chr1         | 44130182  | 44130470  | ...   |\n",
       " | chr1         | 51512257  | 51513447  | ...   |\n",
       " | ...          | ...       | ...       | ...   |\n",
       " | chr19        | 55549844  | 55550530  | ...   |\n",
       " | chr19        | 55549844  | 55550530  | ...   |\n",
       " | chr19        | 47430027  | 47430434  | ...   |\n",
       " | chr19        | 47149908  | 47150175  | ...   |\n",
       " | chrX         | 36126211  | 36126785  | ...   |\n",
       " +--------------+-----------+-----------+-------+\n",
       " Unstranded PyRanges object has 301 rows and 10 columns from 20 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.\n",
       " Considered unstranded due to these Strand values: '.'\n",
       " 7 hidden columns: Name, Score, Strand, ThickStart, ... (+ 3 more.),\n",
       " +--------------+-----------+-----------+\n",
       " | Chromosome   | Start     | End       |\n",
       " | (category)   | (int32)   | (int32)   |\n",
       " |--------------+-----------+-----------|\n",
       " | chr1         | 189908067 | 189908567 |\n",
       " | chr1         | 136834224 | 136834724 |\n",
       " | chr1         | 194393648 | 194394148 |\n",
       " | chr1         | 194373388 | 194373888 |\n",
       " | ...          | ...       | ...       |\n",
       " | chrX         | 12285912  | 12286412  |\n",
       " | chrX         | 140429496 | 140429996 |\n",
       " | chrX         | 105885229 | 105885729 |\n",
       " | chrX         | 141151096 | 141151596 |\n",
       " +--------------+-----------+-----------+\n",
       " Unstranded PyRanges object has 1,324 rows and 3 columns from 20 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.,\n",
       " +--------------+-----------+-----------+\n",
       " | Chromosome   | Start     | End       |\n",
       " | (category)   | (int32)   | (int32)   |\n",
       " |--------------+-----------+-----------|\n",
       " | chr1         | 133364979 | 133365479 |\n",
       " | chr1         | 134574976 | 134575476 |\n",
       " | chr1         | 64754583  | 64755083  |\n",
       " | chr1         | 64787550  | 64788050  |\n",
       " | ...          | ...       | ...       |\n",
       " | chrX         | 169608533 | 169609033 |\n",
       " | chrX         | 142478094 | 142478594 |\n",
       " | chrX         | 143532275 | 143532775 |\n",
       " | chrX         | 11887118  | 11887618  |\n",
       " +--------------+-----------+-----------+\n",
       " Unstranded PyRanges object has 7,675 rows and 3 columns from 20 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.,\n",
       " +--------------+-----------+-----------+\n",
       " | Chromosome   | Start     | End       |\n",
       " | (category)   | (int32)   | (int32)   |\n",
       " |--------------+-----------+-----------|\n",
       " | chr1         | 84395934  | 84396434  |\n",
       " | chr1         | 188119759 | 188120259 |\n",
       " | chr1         | 177717425 | 177717925 |\n",
       " | chr1         | 168680628 | 168681128 |\n",
       " | ...          | ...       | ...       |\n",
       " | chrX         | 160412339 | 160412839 |\n",
       " | chrX         | 103712853 | 103713353 |\n",
       " | chrX         | 51398424  | 51398924  |\n",
       " | chrX         | 161487721 | 161488221 |\n",
       " +--------------+-----------+-----------+\n",
       " Unstranded PyRanges object has 1,495 rows and 3 columns from 20 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.,\n",
       " +--------------+-----------+-----------+\n",
       " | Chromosome   | Start     | End       |\n",
       " | (category)   | (int32)   | (int32)   |\n",
       " |--------------+-----------+-----------|\n",
       " | chr1         | 79850316  | 79850816  |\n",
       " | chr1         | 58212771  | 58213271  |\n",
       " | chr1         | 88192624  | 88193124  |\n",
       " | chr1         | 167617410 | 167617910 |\n",
       " | ...          | ...       | ...       |\n",
       " | chrX         | 139089712 | 139090212 |\n",
       " | chrX         | 139808900 | 139809400 |\n",
       " | chrX         | 152224245 | 152224745 |\n",
       " | chrX         | 47663331  | 47663831  |\n",
       " | chrY         | 1206523   | 1207023   |\n",
       " +--------------+-----------+-----------+\n",
       " Unstranded PyRanges object has 3,511 rows and 3 columns from 21 chromosomes.\n",
       " For printing, the PyRanges was sorted on Chromosome.]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "region_sets['Top20K'] = pr.PyRanges(region_sets['All_peaks'].df.sort_values(by=['Score'], ascending=False)[0:20000])\n",
    "region_sets['Top10K'] = pr.PyRanges(region_sets['All_peaks'].df.sort_values(by=['Score'], ascending=False)[0:10000])\n",
    "region_sets['Top5K'] = pr.PyRanges(region_sets['All_peaks'].df.sort_values(by=['Score'], ascending=False)[0:5000])\n",
    "region_sets['Top1K'] = pr.PyRanges(region_sets['All_peaks'].df.sort_values(by=['Score'], ascending=False)[0:1000])\n",
    "region_sets['PP_ChIP_ov'] = pr.read_bed('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/PP_ChIP_overlap.bed')\n",
    "region_sets['PP'] = pr.read_bed('/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/GEMSTAT/Region_selection_v3/regions/Periportal_clean_v4.bed')\n",
    "region_sets['General'] = pr.read_bed('/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/GEMSTAT/Region_selection_v3/regions/General_clean_v4.bed')\n",
    "region_sets['Pericentral'] = pr.read_bed('/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/GEMSTAT/Region_selection_v3/regions/Pericentral_clean_v4.bed')\n",
    "region_sets['Intermediate'] = pr.read_bed('/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/GEMSTAT/Region_selection_v3/regions/Intermediate_clean_v4.bed')\n",
    "[region_sets[x] for x in region_sets.keys()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ca1d04c8-0a54-49f9-9b0d-257cab7b41f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "region_sets['Top1K'].to_csv('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed', sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f68c9617-d552-4630-bc16-cb0623052885",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-11 19:14:28,323 cisTarget    INFO     Reading cisTarget database\n",
      "2023-06-11 19:16:01,200 cisTarget    INFO     Running cisTarget for All_peaks which has 25238 regions\n",
      "2023-06-11 19:16:12,196 cisTarget    INFO     Annotating motifs for All_peaks\n",
      "2023-06-11 19:16:17,267 cisTarget    INFO     Getting cistromes for All_peaks\n",
      "2023-06-11 19:16:17,716 cisTarget    INFO     Running cisTarget for Top20K which has 21711 regions\n",
      "2023-06-11 19:16:28,155 cisTarget    INFO     Annotating motifs for Top20K\n",
      "2023-06-11 19:16:32,658 cisTarget    INFO     Getting cistromes for Top20K\n",
      "2023-06-11 19:16:33,072 cisTarget    INFO     Running cisTarget for Top10K which has 11404 regions\n",
      "2023-06-11 19:16:40,196 cisTarget    INFO     Annotating motifs for Top10K\n",
      "2023-06-11 19:16:43,133 cisTarget    INFO     Getting cistromes for Top10K\n",
      "2023-06-11 19:16:43,495 cisTarget    INFO     Running cisTarget for Top5K which has 5807 regions\n",
      "2023-06-11 19:16:49,172 cisTarget    INFO     Annotating motifs for Top5K\n",
      "2023-06-11 19:16:51,301 cisTarget    INFO     Getting cistromes for Top5K\n",
      "2023-06-11 19:16:51,545 cisTarget    INFO     Running cisTarget for Top1K which has 1167 regions\n",
      "2023-06-11 19:16:56,568 cisTarget    INFO     Annotating motifs for Top1K\n",
      "2023-06-11 19:16:58,236 cisTarget    INFO     Getting cistromes for Top1K\n",
      "2023-06-11 19:16:58,665 cisTarget    INFO     Running cisTarget for PP_ChIP_ov which has 378 regions\n",
      "2023-06-11 19:17:03,347 cisTarget    INFO     Annotating motifs for PP_ChIP_ov\n",
      "2023-06-11 19:17:04,869 cisTarget    INFO     Getting cistromes for PP_ChIP_ov\n",
      "2023-06-11 19:17:05,189 cisTarget    INFO     Running cisTarget for PP which has 1324 regions\n",
      "2023-06-11 19:17:09,798 cisTarget    INFO     Annotating motifs for PP\n",
      "2023-06-11 19:17:11,362 cisTarget    INFO     Getting cistromes for PP\n",
      "2023-06-11 19:17:11,524 cisTarget    INFO     Running cisTarget for General which has 7675 regions\n",
      "2023-06-11 19:17:17,884 cisTarget    INFO     Annotating motifs for General\n",
      "2023-06-11 19:17:20,493 cisTarget    INFO     Getting cistromes for General\n",
      "2023-06-11 19:17:20,757 cisTarget    INFO     Running cisTarget for Pericentral which has 1495 regions\n",
      "2023-06-11 19:17:25,332 cisTarget    INFO     Annotating motifs for Pericentral\n",
      "2023-06-11 19:17:26,908 cisTarget    INFO     Getting cistromes for Pericentral\n",
      "2023-06-11 19:17:27,002 cisTarget    INFO     Running cisTarget for Intermediate which has 3511 regions\n",
      "2023-06-11 19:17:32,382 cisTarget    INFO     Annotating motifs for Intermediate\n",
      "2023-06-11 19:17:34,345 cisTarget    INFO     Getting cistromes for Intermediate\n",
      "2023-06-11 19:17:34,600 cisTarget    INFO     Done!\n"
     ]
    }
   ],
   "source": [
    "# Load cistarget functions\n",
    "from pycistarget.motif_enrichment_cistarget import *\n",
    "# Preload db, you can also just provide the path to the db. Preloading the database is useful if you want to test different parameters.\n",
    "db = '/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/cbust_motif_clustering_combined/Liver_MO.regions_vs_motifs.rankings.v2.feather'\n",
    "#db = '/staging/leuven/stg_00002/icistarget-data/make_rankings/v9/CTX_mm10/CTX_mm10_SCREEN3_1kb_bg_with_mask/CTX_mm10_SCREEN3_1kb_bg_with_mask.regions_vs_motifs.rankings.feather'                           \n",
    "ctx_db = cisTargetDatabase(db, region_sets)    \n",
    "# Remove cisbp\n",
    "#ctx_db.db_rankings = ctx_db.db_rankings[~ctx_db.db_rankings.index.str.contains(\"cisbp\")]\n",
    "# Remove motifs\n",
    "#import pandas as pd\n",
    "#out = pd.read_csv('/staging/leuven/stg_00002/lcb/cbravo/motif_clustering/RNA_harmony_snn_res_5_clusters/motif_collection_combined_motifs_stamp_and_singlets/promoter_motifs.txt', header=None).iloc[:,0].tolist()\n",
    "#ctx_db.db_rankings = ctx_db.db_rankings.drop(out)\n",
    "# Run (r06i00n12)\n",
    "cistarget_dict = run_cistarget(ctx_db = ctx_db,\n",
    "                               region_sets = region_sets,\n",
    "                               specie = 'mus_musculus',\n",
    "                               auc_threshold = 0.005,\n",
    "                               nes_threshold = 3.0,\n",
    "                               rank_threshold = 0.05,\n",
    "                               annotation = ['Direct_annot', 'Orthology_annot'],\n",
    "                               path_to_motif_annotations = '/staging/leuven/stg_00002/lcb/cbravo/Liver/mm_annotation_v9_clust.tsv',\n",
    "                               n_cpu = 1,\n",
    "                               _temp_dir='/scratch/leuven/313/vsc31305/ray_spill',\n",
    "                               annotation_version = 'v9nr_clust')\n",
    "# Save\n",
    "import pickle\n",
    "with open('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/cisTarget_dict_nes3.0.pkl', 'wb') as f:\n",
    "  pickle.dump(cistarget_dict, f)\n",
    "\n",
    "#cistarget_dict = run_cistarget(ctx_db = ctx_db,\n",
    "#                               region_sets = region_sets,\n",
    "#                               specie = 'mus_musculus',\n",
    "#                               auc_threshold = 0.005,\n",
    "#                               nes_threshold = 0.0,\n",
    "#                               rank_threshold = 0.05,\n",
    "#                               annotation = ['Direct_annot', 'Orthology_annot'],\n",
    "#                               path_to_motif_annotations = '/staging/leuven/stg_00002/lcb/cbravo/Liver/mm_annotation_v9_clust.tsv',\n",
    "#                               n_cpu = 1,\n",
    "#                               _temp_dir='/scratch/leuven/313/vsc31305/ray_spill',\n",
    "#                               annotation_version = 'v9nr_clust')\n",
    "# Save\n",
    "#import pickle\n",
    "#with open('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/cisTarget_dict.pkl', 'wb') as f:\n",
    "#  pickle.dump(cistarget_dict, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f659cb8f-e2f3-430e-ae0b-6072cce2ffc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "infile = open('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/cisTarget_dict_nes3.0.pkl', 'rb')\n",
    "cistarget_dict = pickle.load(infile)\n",
    "infile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "23e958ef-5e4e-4f66-a01a-43189bb2257f",
   "metadata": {},
   "outputs": [],
   "source": [
    "for x in cistarget_dict.keys():\n",
    "    out_file = '/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/ctx_Liver_db/'+ str(x) + '.html'\n",
    "    cistarget_dict[x].motif_enrichment.to_html(open(out_file, 'w'), escape=False, col_space=80)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "009a38a1-2b15-4e34-ac1c-b4f0afbf0065",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th style=\"min-width: 80px;\"></th>\n",
       "      <th style=\"min-width: 80px;\">Logo</th>\n",
       "      <th style=\"min-width: 80px;\">Region_set</th>\n",
       "      <th style=\"min-width: 80px;\">Direct_annot</th>\n",
       "      <th style=\"min-width: 80px;\">Orthology_annot</th>\n",
       "      <th style=\"min-width: 80px;\">NES</th>\n",
       "      <th style=\"min-width: 80px;\">AUC</th>\n",
       "      <th style=\"min-width: 80px;\">Rank_at_max</th>\n",
       "      <th style=\"min-width: 80px;\">Motif_hits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>taipale__ONECUT2_DBD_NNAAAATCRATAWN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__ONECUT2_DBD_NNAAAATCRATAWN.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut2</td>\n",
       "      <td>7.887854</td>\n",
       "      <td>0.025114</td>\n",
       "      <td>23624.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__HNF1B_full_NRTTAATNATTAACN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__HNF1B_full_NRTTAATNATTAACN.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>7.498107</td>\n",
       "      <td>0.023998</td>\n",
       "      <td>24295.0</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_90.8</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_90.8.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.390457</td>\n",
       "      <td>0.023690</td>\n",
       "      <td>5116.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.17</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.17.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Onecut1, Onecut2</td>\n",
       "      <td>Onecut3, Onecut1, Onecut2</td>\n",
       "      <td>6.321689</td>\n",
       "      <td>0.020631</td>\n",
       "      <td>20571.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1632</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1632.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Eomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.214591</td>\n",
       "      <td>0.020324</td>\n",
       "      <td>23475.0</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__NR2F1_NRGGTCRNYGACCYN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__NR2F1_NRGGTCRNYGACCYN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nr2f1</td>\n",
       "      <td>6.124055</td>\n",
       "      <td>0.020065</td>\n",
       "      <td>14800.0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_17.26</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_17.26.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Rxra</td>\n",
       "      <td>Rara, Mecom, Thrb, Nr1h4, Rarb, Rorb, Ppard, Rxrb, Prdm16, Nr2f6, Rarg, Rxrg, Rorc, Nr2c1, Rxra, Nr2f1</td>\n",
       "      <td>5.912068</td>\n",
       "      <td>0.019458</td>\n",
       "      <td>15026.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ONECUT2_NTATTGATTWN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__ONECUT2_NTATTGATTWN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut2</td>\n",
       "      <td>5.838093</td>\n",
       "      <td>0.019247</td>\n",
       "      <td>24284.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__FOXA2_NWNWGTMAATATTKRYNYWN_eDBD_meth_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__FOXA2_NWNWGTMAATATTKRYNYWN_eDBD_meth_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.721611</td>\n",
       "      <td>0.018913</td>\n",
       "      <td>23492.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_83.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_83.3.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nr2f1</td>\n",
       "      <td>5.684623</td>\n",
       "      <td>0.018807</td>\n",
       "      <td>18615.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__TBX6_NAGGTGTGAN_eDBD</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__TBX6_NAGGTGTGAN_eDBD.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tbx6</td>\n",
       "      <td>5.523425</td>\n",
       "      <td>0.018346</td>\n",
       "      <td>14385.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1635</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1635.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.522321</td>\n",
       "      <td>0.018343</td>\n",
       "      <td>24177.0</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4283</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4283.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pknox2, Tgif1, Meis1, Meis3, Pknox1, Tgif2, Tgif2lx1, Tgif2lx2, Meis2</td>\n",
       "      <td>5.402526</td>\n",
       "      <td>0.018000</td>\n",
       "      <td>15750.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>stark__VRGKTYAWTGAMMYY</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/stark__VRGKTYAWTGAMMYY.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rara, Thra, Thrb, Vdr, Nr1h4, Nr1i2, Nr1h5, Nr1h3, Rarg, Nr1h2, Nr1i3, Rarb</td>\n",
       "      <td>5.323031</td>\n",
       "      <td>0.017772</td>\n",
       "      <td>845.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00094</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00094.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pbx1, Nr1h4, Nr1i2, Nr1i3, Rxra</td>\n",
       "      <td>5.315854</td>\n",
       "      <td>0.017752</td>\n",
       "      <td>17478.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1629</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1629.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Tbx5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.247952</td>\n",
       "      <td>0.017557</td>\n",
       "      <td>19349.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00389</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00389.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx</td>\n",
       "      <td>5.231943</td>\n",
       "      <td>0.017512</td>\n",
       "      <td>7633.0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_27.33</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_27.33.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.222558</td>\n",
       "      <td>0.017485</td>\n",
       "      <td>2308.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_public__M00132</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_public__M00132.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.205444</td>\n",
       "      <td>0.017436</td>\n",
       "      <td>17935.0</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00416</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00416.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>5.177842</td>\n",
       "      <td>0.017357</td>\n",
       "      <td>5771.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_38.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_38.12.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Pknox2, Meis1, Meis3, Pknox1, Tgif2</td>\n",
       "      <td>Pknox2, Map4k2, Tgif1, Meis1, Meis3, Pknox1, Tgif2, Tgif2lx2, Tgif2lx1, Meis2</td>\n",
       "      <td>5.173425</td>\n",
       "      <td>0.017344</td>\n",
       "      <td>2979.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__FOXA1_NWRTGTAMAYAN_eDBD_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__FOXA1_NWRTGTAMAYAN_eDBD_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxa1</td>\n",
       "      <td>5.171769</td>\n",
       "      <td>0.017339</td>\n",
       "      <td>10979.0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__Meis2_DBD_NTGACAN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__Meis2_DBD_NTGACAN.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Meis2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.023268</td>\n",
       "      <td>0.016914</td>\n",
       "      <td>21275.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_102.5</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_102.5.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut3, Onecut1, Onecut2</td>\n",
       "      <td>4.996769</td>\n",
       "      <td>0.016838</td>\n",
       "      <td>23278.0</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6048</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6048.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Meis2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.982968</td>\n",
       "      <td>0.016799</td>\n",
       "      <td>17962.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_65.14</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_65.14.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxc2, Foxb2, Foxa2, Foxd2, Foxd3, Foxa3, Foxa1, Foxc1, Foxb1</td>\n",
       "      <td>4.938804</td>\n",
       "      <td>0.016672</td>\n",
       "      <td>14050.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6170</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6170.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpb</td>\n",
       "      <td>4.902921</td>\n",
       "      <td>0.016570</td>\n",
       "      <td>24032.0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6050</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6050.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Meis3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.890776</td>\n",
       "      <td>0.016535</td>\n",
       "      <td>15589.0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__CUX1_SRF_NCCATAYWNGGWNNNNATCRATN_CAP_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__CUX1_SRF_NCCATAYWNGGWNNNNATCRATN_CAP_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Srf, Cux1</td>\n",
       "      <td>4.765461</td>\n",
       "      <td>0.016176</td>\n",
       "      <td>2151.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6171</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6171.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpd</td>\n",
       "      <td>4.663331</td>\n",
       "      <td>0.015884</td>\n",
       "      <td>20340.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.24</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.24.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Cebpe</td>\n",
       "      <td>Cebpg, Myc, Cebpa, Cebpe, Ep300, Bmyc, Cebpb, Ddit3</td>\n",
       "      <td>4.599846</td>\n",
       "      <td>0.015702</td>\n",
       "      <td>7278.0</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Achi_Cell_FBgn0033749</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Achi_Cell_FBgn0033749.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tgif2, Tgif2lx2, Tgif2lx1, Tgif1</td>\n",
       "      <td>4.562306</td>\n",
       "      <td>0.015595</td>\n",
       "      <td>18281.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_21.23</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_21.23.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>Zfp12, Hnf1a, Hnf1b, Zfhx3</td>\n",
       "      <td>4.498821</td>\n",
       "      <td>0.015413</td>\n",
       "      <td>21308.0</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4664</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4664.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxa1</td>\n",
       "      <td>4.428158</td>\n",
       "      <td>0.015211</td>\n",
       "      <td>24168.0</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00371</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00371.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pdx1, Runx3</td>\n",
       "      <td>4.414357</td>\n",
       "      <td>0.015171</td>\n",
       "      <td>2658.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__TFAP2C_MAX_TNSCCNNNGGSNNNNCACGTGN_CAP_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__TFAP2C_MAX_TNSCCNNNGGSNNNNCACGTGN_CAP_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Max, Tfap2c</td>\n",
       "      <td>4.332102</td>\n",
       "      <td>0.014936</td>\n",
       "      <td>1033.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>elemento__GCTGACC</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/elemento__GCTGACC.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.325477</td>\n",
       "      <td>0.014917</td>\n",
       "      <td>891.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__disco-r-F1-2_SOLEXA_FBgn0042650</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__disco-r-F1-2_SOLEXA_FBgn0042650.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.313884</td>\n",
       "      <td>0.014884</td>\n",
       "      <td>24096.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00124</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00124.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf4g, Hnf4a, Nr2f2, Pdx1, Nr2f1</td>\n",
       "      <td>4.278553</td>\n",
       "      <td>0.014783</td>\n",
       "      <td>1861.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>homer__AGGTCANTGACCTN_FXR</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/homer__AGGTCANTGACCTN_FXR.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Nr1h4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.262543</td>\n",
       "      <td>0.014737</td>\n",
       "      <td>2646.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M04653</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M04653.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Foxj3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.258127</td>\n",
       "      <td>0.014724</td>\n",
       "      <td>3988.0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_89.22</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_89.22.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pbx4</td>\n",
       "      <td>4.182496</td>\n",
       "      <td>0.014508</td>\n",
       "      <td>1642.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Hth_Cell_FBgn0001235</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Hth_Cell_FBgn0001235.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Meis2</td>\n",
       "      <td>4.176424</td>\n",
       "      <td>0.014490</td>\n",
       "      <td>11207.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_public__M00220</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_public__M00220.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Srebf1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.174767</td>\n",
       "      <td>0.014485</td>\n",
       "      <td>3547.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_14.61</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_14.61.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tgif2, Tgif2lx2, Tgif2lx1, Tgif1</td>\n",
       "      <td>4.134468</td>\n",
       "      <td>0.014370</td>\n",
       "      <td>16583.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0950.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0950.1.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.123427</td>\n",
       "      <td>0.014339</td>\n",
       "      <td>21697.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.12.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Tcf7l2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.107417</td>\n",
       "      <td>0.014293</td>\n",
       "      <td>23432.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bergman__EcR_usp</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/bergman__EcR_usp.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rara, Thra, Thrb, Vdr, Nr1h4, Nr1i2, Nr1h5, Nr1h3, Rxrb, Rarg, Nr1h2, Nr1i3, Rarb, Rxra, Rxrg</td>\n",
       "      <td>4.073742</td>\n",
       "      <td>0.014196</td>\n",
       "      <td>7798.0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_10.34</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_10.34.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.035099</td>\n",
       "      <td>0.014086</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0102.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0102.3.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpa</td>\n",
       "      <td>4.017433</td>\n",
       "      <td>0.014035</td>\n",
       "      <td>21115.0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_48.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_48.12.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.016329</td>\n",
       "      <td>0.014032</td>\n",
       "      <td>4075.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>elemento__TTGTTGAC</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/elemento__TTGTTGAC.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.987623</td>\n",
       "      <td>0.013950</td>\n",
       "      <td>1001.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_87.35</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_87.35.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.984310</td>\n",
       "      <td>0.013940</td>\n",
       "      <td>1302.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M2339</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M2339.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.982654</td>\n",
       "      <td>0.013936</td>\n",
       "      <td>6231.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_75.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_75.4.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smad4, Bnc2, Bnc1</td>\n",
       "      <td>3.979894</td>\n",
       "      <td>0.013928</td>\n",
       "      <td>1658.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>predrem__nrMotif2417</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/predrem__nrMotif2417.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.971061</td>\n",
       "      <td>0.013902</td>\n",
       "      <td>388.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00210</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00210.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cdx2, Hmbox1</td>\n",
       "      <td>3.965541</td>\n",
       "      <td>0.013887</td>\n",
       "      <td>219.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1925</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1925.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpa</td>\n",
       "      <td>3.965541</td>\n",
       "      <td>0.013887</td>\n",
       "      <td>23138.0</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07968</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M07968.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nr2e1</td>\n",
       "      <td>3.936282</td>\n",
       "      <td>0.013803</td>\n",
       "      <td>1318.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00041</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00041.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx, Hmbox1</td>\n",
       "      <td>3.908679</td>\n",
       "      <td>0.013724</td>\n",
       "      <td>14170.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_38.11</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_38.11.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Tgif1, Meis1, Snai1, Meis3, Tgif2, Meis2</td>\n",
       "      <td>Pknox2, Tgif1, Meis1, Meis3, Pknox1, Tgif2, Tgif2lx1, Tgif2lx2, Meis2</td>\n",
       "      <td>3.905367</td>\n",
       "      <td>0.013714</td>\n",
       "      <td>3257.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_1.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_1.4.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Foxa2, Foxl1, Foxj3, Foxd1, Foxf2, Foxk1, Bbx, Foxa3, Foxa1, Foxc1, Foxl2, Foxb1</td>\n",
       "      <td>Foxf1, Foxj1, Gm43517, Foxn2, Foxd2, Foxf2, Foxa3, Gata3, Foxr1, Foxb1, Tcf12, Bcl3, Foxq1, Foxg1, Foxj3, Foxi2, Mbd4, Mybl2, Foxp1, Gm5294, Foxd1, Foxk1, Nfic, Foxd3, Ep300, Foxo1, Foxi1, Foxp3, Foxj2, Foxp4, Foxc2, Foxa2, Foxp2, Hdac2, Foxk2, Foxi3, Foxn3, Foxa1, Foxc1, Foxl2, Tcf7l2</td>\n",
       "      <td>3.881629</td>\n",
       "      <td>0.013646</td>\n",
       "      <td>20105.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.31</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.31.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.872244</td>\n",
       "      <td>0.013620</td>\n",
       "      <td>21931.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__PPARA_MOUSE.H11MO.1.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__PPARA_MOUSE.H11MO.1.A.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Ppara</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.846850</td>\n",
       "      <td>0.013547</td>\n",
       "      <td>14259.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_38.10</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_38.10.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Six4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.830288</td>\n",
       "      <td>0.013499</td>\n",
       "      <td>5136.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.26</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.26.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Prdm5</td>\n",
       "      <td>Vamp3</td>\n",
       "      <td>3.820352</td>\n",
       "      <td>0.013471</td>\n",
       "      <td>1275.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_89.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_89.4.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Nr2e1</td>\n",
       "      <td>Nr2e1</td>\n",
       "      <td>3.806550</td>\n",
       "      <td>0.013432</td>\n",
       "      <td>2093.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__FOXM1_HUMAN.H11MO.0.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__FOXM1_HUMAN.H11MO.0.A.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxm1</td>\n",
       "      <td>3.799926</td>\n",
       "      <td>0.013413</td>\n",
       "      <td>24205.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00160</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00160.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx, Rax</td>\n",
       "      <td>3.793853</td>\n",
       "      <td>0.013395</td>\n",
       "      <td>23561.0</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_36.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_36.1.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prdm4</td>\n",
       "      <td>3.790541</td>\n",
       "      <td>0.013386</td>\n",
       "      <td>234.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1140</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1140.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pknox2, Tgif1, Meis1, Meis3, Pknox1, Tgif2, Tgif2lx1, Tgif2lx2, Meis2</td>\n",
       "      <td>3.781708</td>\n",
       "      <td>0.013360</td>\n",
       "      <td>1465.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0148.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0148.3.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxa1</td>\n",
       "      <td>3.772323</td>\n",
       "      <td>0.013334</td>\n",
       "      <td>10991.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>neph__UW.Motif.0511</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/neph__UW.Motif.0511.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.717118</td>\n",
       "      <td>0.013176</td>\n",
       "      <td>1253.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_10.137</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_10.137.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.707733</td>\n",
       "      <td>0.013149</td>\n",
       "      <td>1044.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0019.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0019.1.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.672954</td>\n",
       "      <td>0.013049</td>\n",
       "      <td>1624.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M00801</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M00801.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Crem, Atf4, Atf7, Creb1, Atf2, Atf3, Atf1</td>\n",
       "      <td>Gm28047</td>\n",
       "      <td>3.656393</td>\n",
       "      <td>0.013002</td>\n",
       "      <td>24056.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00011</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00011.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxa1, Foxa2, Foxa3</td>\n",
       "      <td>3.651976</td>\n",
       "      <td>0.012989</td>\n",
       "      <td>10938.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__HLH106_SANGER_10_FBgn0015234</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__HLH106_SANGER_10_FBgn0015234.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Srebf2, Srebf1</td>\n",
       "      <td>3.648112</td>\n",
       "      <td>0.012978</td>\n",
       "      <td>7018.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__CUX1_TBX21_TMACACCYAATA_CAP_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__CUX1_TBX21_TMACACCYAATA_CAP_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tbx21, Cux1</td>\n",
       "      <td>3.642591</td>\n",
       "      <td>0.012962</td>\n",
       "      <td>9651.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6282</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6282.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.624926</td>\n",
       "      <td>0.012912</td>\n",
       "      <td>23361.0</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00328</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00328.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nfyc, Nfyb, Nfya, Hnf1b, Hnf1a</td>\n",
       "      <td>3.622718</td>\n",
       "      <td>0.012905</td>\n",
       "      <td>21528.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>predrem__nrMotif1082</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/predrem__nrMotif1082.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.601740</td>\n",
       "      <td>0.012845</td>\n",
       "      <td>1746.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00232</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00232.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpb, Taf7, Tbp</td>\n",
       "      <td>3.590147</td>\n",
       "      <td>0.012812</td>\n",
       "      <td>8425.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_37.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_37.1.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.586834</td>\n",
       "      <td>0.012803</td>\n",
       "      <td>9871.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_39.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_39.3.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Rara, Nr5a2, Esrrg, Esrra, Rarg</td>\n",
       "      <td>Rara, Rora, Esrrg, Esrrb, Esrra, Rarg</td>\n",
       "      <td>3.556472</td>\n",
       "      <td>0.012716</td>\n",
       "      <td>753.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.16.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Pax7, Onecut1, Cux1</td>\n",
       "      <td>Pax3, Onecut1, Cux1, Onecut3, Pax6, Onecut2, Cux2, Pax7</td>\n",
       "      <td>3.538254</td>\n",
       "      <td>0.012664</td>\n",
       "      <td>21409.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__GCM1_HOXA2_RTRSGGGNNTAATKR_CAP_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__GCM1_HOXA2_RTRSGGGNNTAATKR_CAP_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gm28308, Hoxa2, Gcm1</td>\n",
       "      <td>3.521141</td>\n",
       "      <td>0.012615</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__MEIS1_DRGX_TGACASTAATKR_CAP_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__MEIS1_DRGX_TGACASTAATKR_CAP_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Meis1, Prrxl1</td>\n",
       "      <td>3.515068</td>\n",
       "      <td>0.012597</td>\n",
       "      <td>662.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.50</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.50.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.512308</td>\n",
       "      <td>0.012589</td>\n",
       "      <td>1755.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00288</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00288.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hmga1-rs1, Sry, Hmga1, Hmga2</td>\n",
       "      <td>3.494642</td>\n",
       "      <td>0.012539</td>\n",
       "      <td>16849.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M04829</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M04829.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stat3</td>\n",
       "      <td>3.469800</td>\n",
       "      <td>0.012468</td>\n",
       "      <td>16750.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_26.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_26.2.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.433917</td>\n",
       "      <td>0.012365</td>\n",
       "      <td>350.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hdpi__ZFP3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hdpi__ZFP3.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Zfp3</td>\n",
       "      <td>3.431709</td>\n",
       "      <td>0.012359</td>\n",
       "      <td>2626.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_71.28</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_71.28.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prdm4</td>\n",
       "      <td>3.405762</td>\n",
       "      <td>0.012284</td>\n",
       "      <td>2342.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_94.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_94.6.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tbx21</td>\n",
       "      <td>3.405210</td>\n",
       "      <td>0.012283</td>\n",
       "      <td>1493.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_35.17</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_35.17.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Maf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.400794</td>\n",
       "      <td>0.012270</td>\n",
       "      <td>2986.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0208</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0208.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Mitf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.371535</td>\n",
       "      <td>0.012186</td>\n",
       "      <td>821.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_43.19</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_43.19.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pou3f2</td>\n",
       "      <td>3.362702</td>\n",
       "      <td>0.012161</td>\n",
       "      <td>1804.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__ATF4_CEBPD_NGATGATGCAATNN_CAP</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__ATF4_CEBPD_NGATGATGCAATNN_CAP.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpd, Atf4</td>\n",
       "      <td>3.362702</td>\n",
       "      <td>0.012161</td>\n",
       "      <td>7536.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_101.11</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_101.11.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.354422</td>\n",
       "      <td>0.012137</td>\n",
       "      <td>2058.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_26.9</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_26.9.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Zfp775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.350005</td>\n",
       "      <td>0.012125</td>\n",
       "      <td>6989.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_42.46</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_42.46.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Zfp867</td>\n",
       "      <td>3.325163</td>\n",
       "      <td>0.012054</td>\n",
       "      <td>11424.0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.18</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.18.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut3, Onecut1, Onecut2</td>\n",
       "      <td>3.322403</td>\n",
       "      <td>0.012046</td>\n",
       "      <td>21081.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M06279</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M06279.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.319090</td>\n",
       "      <td>0.012036</td>\n",
       "      <td>1787.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_17.22</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_17.22.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.310810</td>\n",
       "      <td>0.012013</td>\n",
       "      <td>999.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NR2E1_HUMAN.H11MO.0.D</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__NR2E1_HUMAN.H11MO.0.D.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nr2e1</td>\n",
       "      <td>3.309706</td>\n",
       "      <td>0.012009</td>\n",
       "      <td>862.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__ATF4_TEF_RNMTGATGCAATN_CAP</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__ATF4_TEF_RNMTGATGCAATN_CAP.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tef, Atf4</td>\n",
       "      <td>3.304185</td>\n",
       "      <td>0.011994</td>\n",
       "      <td>2687.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__SOX21_DBD_AACAATNNNNAKTGTT</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__SOX21_DBD_AACAATNNNNAKTGTT.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sox21</td>\n",
       "      <td>3.301425</td>\n",
       "      <td>0.011986</td>\n",
       "      <td>965.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_90.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_90.4.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Tbx3, Tbx5, Tbx4, Tbx2, T, Tbx20, Eomes, Tbx21, Tbr1</td>\n",
       "      <td>Tbx15, Tbx3, Tbx1, Mga, Tbx18, Tbx5, Tbx6, Tbx4, Tbx2, T, Tbx20, Tbx10, Eomes, Tbx21, Tbr1</td>\n",
       "      <td>3.294800</td>\n",
       "      <td>0.011967</td>\n",
       "      <td>23787.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>predrem__nrMotif1202</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/predrem__nrMotif1202.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.289832</td>\n",
       "      <td>0.011952</td>\n",
       "      <td>4603.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tiffin__TIFDMEM0000027</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tiffin__TIFDMEM0000027.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.274374</td>\n",
       "      <td>0.011908</td>\n",
       "      <td>363.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>predrem__nrMotif879</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/predrem__nrMotif879.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.260573</td>\n",
       "      <td>0.011869</td>\n",
       "      <td>381.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_16.62</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_16.62.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Zfp668</td>\n",
       "      <td>Zfp668</td>\n",
       "      <td>3.240699</td>\n",
       "      <td>0.011812</td>\n",
       "      <td>572.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.26</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.26.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ivd</td>\n",
       "      <td>3.231867</td>\n",
       "      <td>0.011787</td>\n",
       "      <td>22689.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0534.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0534.1.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rxrg, Rxra, Rxrb</td>\n",
       "      <td>3.231867</td>\n",
       "      <td>0.011787</td>\n",
       "      <td>2505.0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4600</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4600.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gata1</td>\n",
       "      <td>3.224138</td>\n",
       "      <td>0.011764</td>\n",
       "      <td>65.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M09306</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M09306.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.223034</td>\n",
       "      <td>0.011761</td>\n",
       "      <td>1046.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_29.36</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_29.36.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.223034</td>\n",
       "      <td>0.011761</td>\n",
       "      <td>1782.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00002</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00002.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx</td>\n",
       "      <td>3.216409</td>\n",
       "      <td>0.011742</td>\n",
       "      <td>22528.0</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00582</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00582.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hic1, Meis1</td>\n",
       "      <td>3.212545</td>\n",
       "      <td>0.011731</td>\n",
       "      <td>2086.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_75.5</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_75.5.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Six4, Six5</td>\n",
       "      <td>3.202056</td>\n",
       "      <td>0.011701</td>\n",
       "      <td>24297.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00353</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00353.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fosb, Fosl1, Jund, Sox4, Junb, Fos, Fosl2, Jun</td>\n",
       "      <td>3.182734</td>\n",
       "      <td>0.011646</td>\n",
       "      <td>7673.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.48</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.48.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.147955</td>\n",
       "      <td>0.011546</td>\n",
       "      <td>272.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>predrem__nrMotif1493</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/predrem__nrMotif1493.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.131946</td>\n",
       "      <td>0.011501</td>\n",
       "      <td>1175.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_55.19</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_55.19.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.126425</td>\n",
       "      <td>0.011485</td>\n",
       "      <td>4687.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6384</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6384.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nr1h4</td>\n",
       "      <td>3.119248</td>\n",
       "      <td>0.011464</td>\n",
       "      <td>5617.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__MLXIPL_full_ATCACGTGAT</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__MLXIPL_full_ATCACGTGAT.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mlxipl</td>\n",
       "      <td>3.118696</td>\n",
       "      <td>0.011463</td>\n",
       "      <td>1835.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_46.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_46.16.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.116488</td>\n",
       "      <td>0.011456</td>\n",
       "      <td>2325.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_40.13</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_40.13.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Usf2</td>\n",
       "      <td>3.108207</td>\n",
       "      <td>0.011433</td>\n",
       "      <td>866.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__HOXB2_TCF3_NCACCTGNNNNNMATTA_CAP_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__HOXB2_TCF3_NCACCTGNNNNNMATTA_CAP_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tcf3, Hoxb2</td>\n",
       "      <td>3.107103</td>\n",
       "      <td>0.011429</td>\n",
       "      <td>261.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.27</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.27.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Plagl1, Pax3, Phox2a</td>\n",
       "      <td>3.103791</td>\n",
       "      <td>0.011420</td>\n",
       "      <td>934.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00546</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00546.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxc8, Nfe2l1</td>\n",
       "      <td>3.096614</td>\n",
       "      <td>0.011399</td>\n",
       "      <td>17873.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.13</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.13.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Nobox</td>\n",
       "      <td>Lhx9, Dlx4, Hesx1, Dlx1, Dlx6</td>\n",
       "      <td>3.094406</td>\n",
       "      <td>0.011393</td>\n",
       "      <td>1956.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__TBX3_MOUSE.H11MO.0.B</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__TBX3_MOUSE.H11MO.0.B.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Tbx3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.081157</td>\n",
       "      <td>0.011355</td>\n",
       "      <td>10595.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0846</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0846.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.080605</td>\n",
       "      <td>0.011354</td>\n",
       "      <td>22921.0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__FOXB1_full_WGTAAAYAN_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__FOXB1_full_WGTAAAYAN_repr.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxb1</td>\n",
       "      <td>3.076741</td>\n",
       "      <td>0.011343</td>\n",
       "      <td>980.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_39.26</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_39.26.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.072324</td>\n",
       "      <td>0.011330</td>\n",
       "      <td>3579.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M09100</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M09100.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.068460</td>\n",
       "      <td>0.011319</td>\n",
       "      <td>998.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_14.10</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_14.10.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Six2</td>\n",
       "      <td>Six1</td>\n",
       "      <td>3.045274</td>\n",
       "      <td>0.011252</td>\n",
       "      <td>1555.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M00762</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M00762.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>Hnf4g, Hnf4a, Rxra, Nr2f2, Pparg, Nr2f1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029816</td>\n",
       "      <td>0.011208</td>\n",
       "      <td>1999.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ONECUT1_NTATTGATCSGN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__ONECUT1_NTATTGATCSGN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut1</td>\n",
       "      <td>3.028712</td>\n",
       "      <td>0.011205</td>\n",
       "      <td>24234.0</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_31.41</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_31.41.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.025952</td>\n",
       "      <td>0.011197</td>\n",
       "      <td>1006.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.60</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.60.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.007182</td>\n",
       "      <td>0.011143</td>\n",
       "      <td>302.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>predrem__nrMotif1172</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/predrem__nrMotif1172.png\" width=\"200\" ></td>\n",
       "      <td>PP_ChIP_ov</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.005526</td>\n",
       "      <td>0.011139</td>\n",
       "      <td>4342.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pycistarget.motif_enrichment_cistarget import *\n",
    "cistarget_results(cistarget_dict, name='PP_ChIP_ov')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d41353a1-e70e-4f26-b561-0d6b42df66ec",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th style=\"min-width: 80px;\"></th>\n",
       "      <th style=\"min-width: 80px;\">Logo</th>\n",
       "      <th style=\"min-width: 80px;\">Region_set</th>\n",
       "      <th style=\"min-width: 80px;\">Direct_annot</th>\n",
       "      <th style=\"min-width: 80px;\">Orthology_annot</th>\n",
       "      <th style=\"min-width: 80px;\">NES</th>\n",
       "      <th style=\"min-width: 80px;\">AUC</th>\n",
       "      <th style=\"min-width: 80px;\">Rank_at_max</th>\n",
       "      <th style=\"min-width: 80px;\">Motif_hits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>cisbp__M6170</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6170.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpb</td>\n",
       "      <td>9.427588</td>\n",
       "      <td>0.023952</td>\n",
       "      <td>24335.0</td>\n",
       "      <td>227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M03571</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M03571.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Cebpd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.630414</td>\n",
       "      <td>0.019983</td>\n",
       "      <td>22100.0</td>\n",
       "      <td>209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6171</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6171.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpd</td>\n",
       "      <td>7.122160</td>\n",
       "      <td>0.018861</td>\n",
       "      <td>23018.0</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.4.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Dbp, Hlf, Cebpg, Cebpa, Cebpe, Cebpd, Cebpb</td>\n",
       "      <td>Ppargc1a, Dbp, Hsf1, Cebpg, Nfil3, Cebpa, Cebpe, Ep300, Cebpd, Cebpb</td>\n",
       "      <td>6.906513</td>\n",
       "      <td>0.018384</td>\n",
       "      <td>22818.0</td>\n",
       "      <td>215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__CEBPE_DBD_NTTRCGCAAY</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__CEBPE_DBD_NTTRCGCAAY.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpe</td>\n",
       "      <td>6.435166</td>\n",
       "      <td>0.017343</td>\n",
       "      <td>23943.0</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0102.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0102.3.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpa</td>\n",
       "      <td>6.289029</td>\n",
       "      <td>0.017021</td>\n",
       "      <td>22838.0</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M3036</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M3036.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpb</td>\n",
       "      <td>6.247488</td>\n",
       "      <td>0.016929</td>\n",
       "      <td>24225.0</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00041</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00041.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx, Hmbox1</td>\n",
       "      <td>6.217533</td>\n",
       "      <td>0.016863</td>\n",
       "      <td>16459.0</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1925</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1925.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpa</td>\n",
       "      <td>6.065603</td>\n",
       "      <td>0.016527</td>\n",
       "      <td>24308.0</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__HNF1B_full_NRTTAATNATTAACN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__HNF1B_full_NRTTAATNATTAACN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>5.729636</td>\n",
       "      <td>0.015785</td>\n",
       "      <td>23078.0</td>\n",
       "      <td>165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__EP300__ENCSR000EDV_1__m2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__EP300__ENCSR000EDV_1__m2.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ep300</td>\n",
       "      <td>5.703156</td>\n",
       "      <td>0.015727</td>\n",
       "      <td>22883.0</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>homer__NTATYGATCH_HNF6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/homer__NTATYGATCH_HNF6.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Onecut1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.622722</td>\n",
       "      <td>0.015549</td>\n",
       "      <td>23888.0</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M2339</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M2339.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.442161</td>\n",
       "      <td>0.015150</td>\n",
       "      <td>24001.0</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__NFYB__ENCSR000DNM_1__m2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__NFYB__ENCSR000DNM_1__m2.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nfyb</td>\n",
       "      <td>5.353783</td>\n",
       "      <td>0.014955</td>\n",
       "      <td>7940.0</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__FOXA2_NWNWGTMAATATTKRYNYWN_eDBD_meth_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__FOXA2_NWNWGTMAATATTKRYNYWN_eDBD_meth_repr.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.279804</td>\n",
       "      <td>0.014792</td>\n",
       "      <td>10356.0</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ONECUT1_NYATCGATYN_eDBD</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__ONECUT1_NYATCGATYN_eDBD.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut1</td>\n",
       "      <td>5.053730</td>\n",
       "      <td>0.014292</td>\n",
       "      <td>22904.0</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07037</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M07037.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Cebpa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.921495</td>\n",
       "      <td>0.014000</td>\n",
       "      <td>24227.0</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__MGA_DLX3_AGGTGNTAATKR_CAP</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__MGA_DLX3_AGGTGNTAATKR_CAP.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mga, Dlx3</td>\n",
       "      <td>4.714950</td>\n",
       "      <td>0.013544</td>\n",
       "      <td>19048.0</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_41.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_41.16.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.685325</td>\n",
       "      <td>0.013479</td>\n",
       "      <td>24176.0</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__ALX4_TBX21_RGGTGYTAATWR_CAP</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__ALX4_TBX21_RGGTGYTAATWR_CAP.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tbx21, Alx4</td>\n",
       "      <td>4.597444</td>\n",
       "      <td>0.013285</td>\n",
       "      <td>23532.0</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_14.14</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_14.14.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hnf4a</td>\n",
       "      <td>Mybl2, Hnf4g, Hnf4a</td>\n",
       "      <td>4.284317</td>\n",
       "      <td>0.012593</td>\n",
       "      <td>23291.0</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.25</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.25.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxc4, Hoxd4, Hoxb4, Hoxa4</td>\n",
       "      <td>4.283489</td>\n",
       "      <td>0.012591</td>\n",
       "      <td>5502.0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_1.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_1.4.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Foxa2, Foxl1, Foxj3, Foxd1, Foxf2, Foxk1, Bbx, Foxa3, Foxa1, Foxc1, Foxl2, Foxb1</td>\n",
       "      <td>Foxf1, Foxj1, Gm43517, Foxn2, Foxd2, Foxf2, Foxa3, Gata3, Foxr1, Foxb1, Tcf12, Bcl3, Foxq1, Foxg1, Foxj3, Foxi2, Mbd4, Mybl2, Foxp1, Gm5294, Foxd1, Foxk1, Nfic, Foxd3, Ep300, Foxo1, Foxi1, Foxp3, Foxj2, Foxp4, Foxc2, Foxa2, Foxp2, Hdac2, Foxk2, Foxi3, Foxn3, Foxa1, Foxc1, Foxl2, Tcf7l2</td>\n",
       "      <td>4.258002</td>\n",
       "      <td>0.012535</td>\n",
       "      <td>21776.0</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__HXA5_HUMAN.H11MO.0.D</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__HXA5_HUMAN.H11MO.0.D.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxa5</td>\n",
       "      <td>4.219109</td>\n",
       "      <td>0.012449</td>\n",
       "      <td>1564.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6292</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6292.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxa5</td>\n",
       "      <td>4.162177</td>\n",
       "      <td>0.012323</td>\n",
       "      <td>1638.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_65.14</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_65.14.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxc2, Foxb2, Foxa2, Foxd2, Foxd3, Foxa3, Foxa1, Foxc1, Foxb1</td>\n",
       "      <td>4.147447</td>\n",
       "      <td>0.012291</td>\n",
       "      <td>24025.0</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NRF1_MOUSE.H11MO.0.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__NRF1_MOUSE.H11MO.0.A.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Nrf1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.144137</td>\n",
       "      <td>0.012283</td>\n",
       "      <td>2712.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5315</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5315.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpb</td>\n",
       "      <td>4.130732</td>\n",
       "      <td>0.012254</td>\n",
       "      <td>23167.0</td>\n",
       "      <td>202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00002</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00002.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx</td>\n",
       "      <td>4.128746</td>\n",
       "      <td>0.012249</td>\n",
       "      <td>23898.0</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NRF1_HUMAN.H11MO.0.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__NRF1_HUMAN.H11MO.0.A.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nrf1</td>\n",
       "      <td>4.126098</td>\n",
       "      <td>0.012244</td>\n",
       "      <td>1447.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01455</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M01455.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>En2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.117657</td>\n",
       "      <td>0.012225</td>\n",
       "      <td>4670.0</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6114</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6114.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxa1</td>\n",
       "      <td>4.108886</td>\n",
       "      <td>0.012205</td>\n",
       "      <td>23145.0</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>homer__CTGCGCATGCGC_NRF1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/homer__CTGCGCATGCGC_NRF1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nrf1</td>\n",
       "      <td>4.064366</td>\n",
       "      <td>0.012107</td>\n",
       "      <td>23678.0</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M05345</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M05345.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Tbx6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.058573</td>\n",
       "      <td>0.012094</td>\n",
       "      <td>6452.0</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.17</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.17.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Onecut1, Onecut2</td>\n",
       "      <td>Onecut3, Onecut1, Onecut2</td>\n",
       "      <td>4.049967</td>\n",
       "      <td>0.012075</td>\n",
       "      <td>22283.0</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5605</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5605.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lmx1b</td>\n",
       "      <td>4.046161</td>\n",
       "      <td>0.012067</td>\n",
       "      <td>6167.0</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__TAF1__ENCSR000BIB_1__m1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__TAF1__ENCSR000BIB_1__m1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Taf1</td>\n",
       "      <td>4.033914</td>\n",
       "      <td>0.012040</td>\n",
       "      <td>917.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__ONECUT2_DBD_NNAAAATCRATAWN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__ONECUT2_DBD_NNAAAATCRATAWN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut2</td>\n",
       "      <td>4.030604</td>\n",
       "      <td>0.012033</td>\n",
       "      <td>24153.0</td>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0919.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0919.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.022329</td>\n",
       "      <td>0.012014</td>\n",
       "      <td>16558.0</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M05300</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M05300.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Mga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.015047</td>\n",
       "      <td>0.011998</td>\n",
       "      <td>7199.0</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ONECUT1_NTATCGATCGNN_FL</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__ONECUT1_NTATCGATCGNN_FL.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut1</td>\n",
       "      <td>3.996345</td>\n",
       "      <td>0.011957</td>\n",
       "      <td>4341.0</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__HLF_NRTTRYGYAAYN_FL_meth_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__HLF_NRTTRYGYAAYN_FL_meth_repr.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.985256</td>\n",
       "      <td>0.011932</td>\n",
       "      <td>21616.0</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01376</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M01376.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Prrx2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.974002</td>\n",
       "      <td>0.011908</td>\n",
       "      <td>1497.0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0221.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0221.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Evx1, Evx2</td>\n",
       "      <td>3.968872</td>\n",
       "      <td>0.011896</td>\n",
       "      <td>16498.0</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4603</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4603.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yy1</td>\n",
       "      <td>3.958942</td>\n",
       "      <td>0.011874</td>\n",
       "      <td>1696.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.16.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Pax7, Onecut1, Cux1</td>\n",
       "      <td>Pax3, Onecut1, Cux1, Onecut3, Pax6, Onecut2, Cux2, Pax7</td>\n",
       "      <td>3.921539</td>\n",
       "      <td>0.011792</td>\n",
       "      <td>20854.0</td>\n",
       "      <td>141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.6.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Nfil3, Tef, Dbp, Hlf</td>\n",
       "      <td>Nfil3, Tef, Dbp, Hlf</td>\n",
       "      <td>3.920049</td>\n",
       "      <td>0.011788</td>\n",
       "      <td>22963.0</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_public__M00099</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_public__M00099.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Prrx2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.895720</td>\n",
       "      <td>0.011735</td>\n",
       "      <td>1013.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__HOXB2_TBX3_AGGTGTTAATKN_CAP</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__HOXB2_TBX3_AGGTGTTAATKN_CAP.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxb2, Tbx3</td>\n",
       "      <td>3.888107</td>\n",
       "      <td>0.011718</td>\n",
       "      <td>21262.0</td>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4741</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4741.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lhx9, Lhx2</td>\n",
       "      <td>3.877681</td>\n",
       "      <td>0.011695</td>\n",
       "      <td>24297.0</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_66.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_66.12.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Tcf7l2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.853187</td>\n",
       "      <td>0.011641</td>\n",
       "      <td>24053.0</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_21.23</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_21.23.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>Zfp12, Hnf1a, Hnf1b, Zfhx3</td>\n",
       "      <td>3.807839</td>\n",
       "      <td>0.011541</td>\n",
       "      <td>23230.0</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00160</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00160.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx, Rax</td>\n",
       "      <td>3.801716</td>\n",
       "      <td>0.011527</td>\n",
       "      <td>23620.0</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_43.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_43.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.771098</td>\n",
       "      <td>0.011459</td>\n",
       "      <td>3881.0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__LHX2_MOUSE.H11MO.0.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__LHX2_MOUSE.H11MO.0.A.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Lhx2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.760341</td>\n",
       "      <td>0.011436</td>\n",
       "      <td>536.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.24</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.24.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Pdx1, Lmx1a, Lmx1b</td>\n",
       "      <td>Hoxa5, Gsx2, Arx, Gsx1</td>\n",
       "      <td>3.741970</td>\n",
       "      <td>0.011395</td>\n",
       "      <td>22642.0</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ONECUT2_NTATTGATTWN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__ONECUT2_NTATTGATTWN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut2</td>\n",
       "      <td>3.697119</td>\n",
       "      <td>0.011296</td>\n",
       "      <td>17238.0</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.15</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.15.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Pax4, Hoxd3, Lhx4, Alx3, Hoxa5, Prrx1, Hoxb6, Bsx, Pdx1, Pax7, Evx2, Hoxa7, Hoxa2, Rax, Lhx6, Hoxa1, Nkx1-1, Isx, Gsx2, Gbx2, Msx2, Hoxd1, Evx1, Vsx2, Gbx1, Vax2, Barx1, En1, Vax1, Hoxc8, Barx2, Hoxb7, Lbx2, Shox2, Hoxa6, Mnx1, Lhx8, Hoxb5, Esx1, Alx1, Hoxa3, Dlx5, Hoxb3, Lhx2, Meox1, Alx4, Phox2a, Nkx1-2, Arx, Nobox</td>\n",
       "      <td>Lhx4, Uncx, Emx2, Lhx1, Gm28308, Lhx6, Skor1, Gbx2, Vsx2, Gbx1, Zbtb49, Vax2, En1, Lhx9, Mnx1, Vsx1, Lhx2, Lhx5, En2</td>\n",
       "      <td>3.695795</td>\n",
       "      <td>0.011293</td>\n",
       "      <td>24221.0</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1097</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1097.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lhx6, Lhx8</td>\n",
       "      <td>3.690168</td>\n",
       "      <td>0.011281</td>\n",
       "      <td>9380.0</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4852</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4852.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.689506</td>\n",
       "      <td>0.011279</td>\n",
       "      <td>1193.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__FOXM1_HUMAN.H11MO.0.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__FOXM1_HUMAN.H11MO.0.A.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxm1</td>\n",
       "      <td>3.656903</td>\n",
       "      <td>0.011207</td>\n",
       "      <td>23793.0</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>scertf__foat.FKH1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/scertf__foat.FKH1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxp4, Foxp3, Foxp2, Foxp1</td>\n",
       "      <td>3.649621</td>\n",
       "      <td>0.011191</td>\n",
       "      <td>23049.0</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__EMX1_HUMAN.H11MO.0.D</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__EMX1_HUMAN.H11MO.0.D.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Emx1</td>\n",
       "      <td>3.636546</td>\n",
       "      <td>0.011162</td>\n",
       "      <td>810.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_19.25</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_19.25.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.634394</td>\n",
       "      <td>0.011158</td>\n",
       "      <td>23961.0</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Awh_Cell_FBgn0013751</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Awh_Cell_FBgn0013751.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lhx6, Lhx8</td>\n",
       "      <td>3.628767</td>\n",
       "      <td>0.011145</td>\n",
       "      <td>9480.0</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__CEBPG_NRTTGCGYAAYN_FL</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__CEBPG_NRTTGCGYAAYN_FL.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpg</td>\n",
       "      <td>3.620823</td>\n",
       "      <td>0.011128</td>\n",
       "      <td>24196.0</td>\n",
       "      <td>163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07801</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M07801.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nkx6-3, Nkx6-2, Nkx6-1</td>\n",
       "      <td>3.605597</td>\n",
       "      <td>0.011094</td>\n",
       "      <td>22274.0</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ONECUT1_NTATTGATCSGN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__ONECUT1_NTATTGATCSGN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut1</td>\n",
       "      <td>3.602453</td>\n",
       "      <td>0.011087</td>\n",
       "      <td>8134.0</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M2030</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M2030.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Evx1, Evx2</td>\n",
       "      <td>3.592026</td>\n",
       "      <td>0.011064</td>\n",
       "      <td>20581.0</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0959</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0959.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Prrx1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.580607</td>\n",
       "      <td>0.011039</td>\n",
       "      <td>20462.0</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4957</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4957.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Evx1, Evx2</td>\n",
       "      <td>3.580110</td>\n",
       "      <td>0.011038</td>\n",
       "      <td>13637.0</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0882</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0882.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.569849</td>\n",
       "      <td>0.011015</td>\n",
       "      <td>2377.0</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M3853</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M3853.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Prrx2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.561243</td>\n",
       "      <td>0.010996</td>\n",
       "      <td>1291.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__HOXB6_YTAATTRY_eDBD_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__HOXB6_YTAATTRY_eDBD_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.558264</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>18646.0</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M2286</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M2286.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf4g</td>\n",
       "      <td>3.555616</td>\n",
       "      <td>0.010984</td>\n",
       "      <td>23253.0</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_82.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_82.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Lhx9, Prrx2, Vsx1, Alx1, Pax6, Lhx2, Alx4, Dlx3, Arx, Shox2</td>\n",
       "      <td>Lmx1b, Lmx1a, Uncx, Alx3, Msx1, Prrx1, Bsx, Otp, Isl2, Evx2, Rax, Isx, Isl1, Mixl1, Gbx2, Pou4f1, Evx1, Vsx2, Noto, En1, Lbx2, Shox2, Lhx9, Prrx2, Lbx1, Hoxb5, Vsx1, Tlx2, Gsx1, Arx, En2</td>\n",
       "      <td>3.545852</td>\n",
       "      <td>0.010962</td>\n",
       "      <td>24331.0</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5319</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5319.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cebpg</td>\n",
       "      <td>3.535425</td>\n",
       "      <td>0.010939</td>\n",
       "      <td>24230.0</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0923.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0923.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.519868</td>\n",
       "      <td>0.010905</td>\n",
       "      <td>11111.0</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__PRRX2_full_NYAATTAN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__PRRX2_full_NYAATTAN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prrx2</td>\n",
       "      <td>3.514406</td>\n",
       "      <td>0.010893</td>\n",
       "      <td>23753.0</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>factorbook__SP1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/factorbook__SP1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sp2, Sp1</td>\n",
       "      <td>3.510765</td>\n",
       "      <td>0.010885</td>\n",
       "      <td>11163.0</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01898</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M01898.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.509110</td>\n",
       "      <td>0.010881</td>\n",
       "      <td>591.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4664</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4664.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxa1</td>\n",
       "      <td>3.493222</td>\n",
       "      <td>0.010846</td>\n",
       "      <td>24134.0</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Eve_SOLEXA_FBgn0000606</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Eve_SOLEXA_FBgn0000606.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Evx1, Evx2</td>\n",
       "      <td>3.492229</td>\n",
       "      <td>0.010844</td>\n",
       "      <td>14578.0</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5542</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5542.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxb3</td>\n",
       "      <td>3.492229</td>\n",
       "      <td>0.010844</td>\n",
       "      <td>17791.0</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_60.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_60.6.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pou4f1, Lbx2, Pou2f1</td>\n",
       "      <td>3.486768</td>\n",
       "      <td>0.010832</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00024</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00024.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pax4, Foxo1</td>\n",
       "      <td>3.461777</td>\n",
       "      <td>0.010776</td>\n",
       "      <td>23037.0</td>\n",
       "      <td>123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6033</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6033.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hoxd3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.450357</td>\n",
       "      <td>0.010751</td>\n",
       "      <td>24293.0</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__POLR2A__ENCSR000DKM_1__m2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__POLR2A__ENCSR000DKM_1__m2.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.435959</td>\n",
       "      <td>0.010719</td>\n",
       "      <td>19826.0</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0921.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0921.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.434469</td>\n",
       "      <td>0.010716</td>\n",
       "      <td>24316.0</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4944</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4944.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>En2, En1</td>\n",
       "      <td>3.416264</td>\n",
       "      <td>0.010676</td>\n",
       "      <td>23138.0</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.16.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Prrx1</td>\n",
       "      <td>Lbx1, Evx1, Evx2</td>\n",
       "      <td>3.410803</td>\n",
       "      <td>0.010664</td>\n",
       "      <td>21565.0</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5117</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5117.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.408817</td>\n",
       "      <td>0.010659</td>\n",
       "      <td>2141.0</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01685</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M01685.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.406500</td>\n",
       "      <td>0.010654</td>\n",
       "      <td>21421.0</td>\n",
       "      <td>114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0881</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0881.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.404679</td>\n",
       "      <td>0.010650</td>\n",
       "      <td>19760.0</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_90.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_90.3.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Tbx21, Tbx1, Tbx20</td>\n",
       "      <td>Tbx15, Tbx3, Tbx1, Tbx22, Tbx2, T, Tbx20, Tbx19, Eomes, Tbx10, Tbr1</td>\n",
       "      <td>3.401369</td>\n",
       "      <td>0.010643</td>\n",
       "      <td>23442.0</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_29.59</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_29.59.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Zscan10, Nrf1</td>\n",
       "      <td>E2f6, Nrf1</td>\n",
       "      <td>3.391108</td>\n",
       "      <td>0.010620</td>\n",
       "      <td>2347.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Btn_Cell_FBgn0014949</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Btn_Cell_FBgn0014949.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Meox2, Meox1</td>\n",
       "      <td>3.389784</td>\n",
       "      <td>0.010617</td>\n",
       "      <td>3678.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__VAX2_DBD_YTAATTAN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__VAX2_DBD_YTAATTAN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vax2</td>\n",
       "      <td>3.388129</td>\n",
       "      <td>0.010614</td>\n",
       "      <td>17968.0</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__MEOX2_NYAATTAN_FL</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__MEOX2_NYAATTAN_FL.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Meox2</td>\n",
       "      <td>3.377703</td>\n",
       "      <td>0.010591</td>\n",
       "      <td>24153.0</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__TYY1_HUMAN.H11MO.0.A</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__TYY1_HUMAN.H11MO.0.A.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yy1</td>\n",
       "      <td>3.363304</td>\n",
       "      <td>0.010559</td>\n",
       "      <td>1812.0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4756</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4756.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lhx6, Lhx8</td>\n",
       "      <td>3.330535</td>\n",
       "      <td>0.010486</td>\n",
       "      <td>13328.0</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__VSX2_DBD_YTAATTAN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__VSX2_DBD_YTAATTAN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vsx2</td>\n",
       "      <td>3.329211</td>\n",
       "      <td>0.010484</td>\n",
       "      <td>21376.0</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Awh_SOLEXA_FBgn0013751</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Awh_SOLEXA_FBgn0013751.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lhx6, Lhx8</td>\n",
       "      <td>3.325901</td>\n",
       "      <td>0.010476</td>\n",
       "      <td>13231.0</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__ALX3_DBD_NNYAATTANN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__ALX3_DBD_NNYAATTANN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Alx3</td>\n",
       "      <td>3.325570</td>\n",
       "      <td>0.010475</td>\n",
       "      <td>24134.0</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__NK7.1_Cell_FBgn0024321</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__NK7.1_Cell_FBgn0024321.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.325404</td>\n",
       "      <td>0.010475</td>\n",
       "      <td>1920.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_5.61</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_5.61.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Sp4</td>\n",
       "      <td>Sp2, Klf15, Klf13, Sp4, Sp3, Sp1</td>\n",
       "      <td>3.325404</td>\n",
       "      <td>0.010475</td>\n",
       "      <td>1183.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5526</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5526.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf4a</td>\n",
       "      <td>3.319446</td>\n",
       "      <td>0.010462</td>\n",
       "      <td>23900.0</td>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5180</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5180.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxc5, Hoxa5, Hoxb5</td>\n",
       "      <td>3.316467</td>\n",
       "      <td>0.010455</td>\n",
       "      <td>11198.0</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M2247</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M2247.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sp2, Sp3, Sp4, Sp1</td>\n",
       "      <td>3.313819</td>\n",
       "      <td>0.010450</td>\n",
       "      <td>915.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0256.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0256.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.311999</td>\n",
       "      <td>0.010446</td>\n",
       "      <td>6535.0</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>elemento__CCGCGGC</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/elemento__CCGCGGC.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.305875</td>\n",
       "      <td>0.010432</td>\n",
       "      <td>2449.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__HNF4A__ENCSR000BLF_1__m1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__HNF4A__ENCSR000BLF_1__m1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf4a</td>\n",
       "      <td>3.303889</td>\n",
       "      <td>0.010428</td>\n",
       "      <td>22678.0</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>elemento__ACGTGAC</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/elemento__ACGTGAC.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.303724</td>\n",
       "      <td>0.010427</td>\n",
       "      <td>17525.0</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__TBX18_NRAGGTGTGAAN_eDBD_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__TBX18_NRAGGTGTGAAN_eDBD_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.298097</td>\n",
       "      <td>0.010415</td>\n",
       "      <td>24201.0</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07838</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M07838.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vsx2</td>\n",
       "      <td>3.287008</td>\n",
       "      <td>0.010390</td>\n",
       "      <td>786.0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__btd_NAR_FBgn0000233</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__btd_NAR_FBgn0000233.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sp2, Sp3, Sp4, Sp1</td>\n",
       "      <td>3.286512</td>\n",
       "      <td>0.010389</td>\n",
       "      <td>1345.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M09562</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M09562.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.271948</td>\n",
       "      <td>0.010357</td>\n",
       "      <td>12542.0</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__HOXA7_NYMATTAN_HT</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__HOXA7_NYMATTAN_HT.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxa7</td>\n",
       "      <td>3.254735</td>\n",
       "      <td>0.010319</td>\n",
       "      <td>3239.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.24</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.24.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Cebpe</td>\n",
       "      <td>Cebpg, Myc, Cebpa, Cebpe, Ep300, Bmyc, Cebpb, Ddit3</td>\n",
       "      <td>3.242157</td>\n",
       "      <td>0.010291</td>\n",
       "      <td>24118.0</td>\n",
       "      <td>190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0962</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0962.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Lhx6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.239509</td>\n",
       "      <td>0.010285</td>\n",
       "      <td>21003.0</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01422</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M01422.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Lhx6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.224614</td>\n",
       "      <td>0.010253</td>\n",
       "      <td>2513.0</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_15.8</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_15.8.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hoxd3, Nanog, Nkx6-1</td>\n",
       "      <td>Lhx4, Uncx, Tlx1, Emx2, Hoxa4, Hoxb6, Tlx3, Evx2, Meox2, Hoxc6, Gm28308, Hoxa7, Hoxa2, Hoxb2, Hoxd4, Emx1, Evx1, Hoxb4, Vsx2, Hoxc4, En1, Hoxb7, Pou2f1, Lbx1, Hoxa6, Pou2f2 , Hoxb5, Tlx2, Meox1, Hoxb3, Lhx2, Lhx3, En2, Pou2f3</td>\n",
       "      <td>3.219815</td>\n",
       "      <td>0.010242</td>\n",
       "      <td>20137.0</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__HOXA1_NTAATTAN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__HOXA1_NTAATTAN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.219484</td>\n",
       "      <td>0.010241</td>\n",
       "      <td>20127.0</td>\n",
       "      <td>104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__LHX6_MOUSE.H11MO.0.C</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/hocomoco__LHX6_MOUSE.H11MO.0.C.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Lhx6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.217332</td>\n",
       "      <td>0.010236</td>\n",
       "      <td>7997.0</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_73.10</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_73.10.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Adnp2</td>\n",
       "      <td>Rreb1</td>\n",
       "      <td>3.215677</td>\n",
       "      <td>0.010233</td>\n",
       "      <td>1700.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_15.9</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_15.9.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Arid3a</td>\n",
       "      <td>Lhx5, Prrxl1, Lhx1</td>\n",
       "      <td>3.214684</td>\n",
       "      <td>0.010231</td>\n",
       "      <td>3925.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_15.11</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_15.11.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Pdx1, Isl1, Lhx3</td>\n",
       "      <td>Evx2, Evx1, Pou6f1, Barx2, Pou3f1</td>\n",
       "      <td>3.211374</td>\n",
       "      <td>0.010223</td>\n",
       "      <td>7905.0</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_90.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_90.4.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Tbx3, Tbx5, Tbx4, Tbx2, T, Tbx20, Eomes, Tbx21, Tbr1</td>\n",
       "      <td>Tbx15, Tbx3, Tbx1, Mga, Tbx18, Tbx5, Tbx6, Tbx4, Tbx2, T, Tbx20, Tbx10, Eomes, Tbx21, Tbr1</td>\n",
       "      <td>3.190356</td>\n",
       "      <td>0.010177</td>\n",
       "      <td>24268.0</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_43.20</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_43.20.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Barhl1, Nobox</td>\n",
       "      <td>Msx1, Barhl2, Msx3, En2, Tlx3, Prrx1, En1, Msx2, Barhl1, Gbx2</td>\n",
       "      <td>3.188535</td>\n",
       "      <td>0.010173</td>\n",
       "      <td>752.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__CG4136_Cell_FBgn0029775</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__CG4136_Cell_FBgn0029775.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vsx2</td>\n",
       "      <td>3.187873</td>\n",
       "      <td>0.010171</td>\n",
       "      <td>4422.0</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4493</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M4493.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yy1</td>\n",
       "      <td>3.187377</td>\n",
       "      <td>0.010170</td>\n",
       "      <td>1451.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_public__M00132</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_public__M00132.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.173971</td>\n",
       "      <td>0.010141</td>\n",
       "      <td>20905.0</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5334</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5334.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.169337</td>\n",
       "      <td>0.010130</td>\n",
       "      <td>23314.0</td>\n",
       "      <td>141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M04869</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M04869.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Egr1</td>\n",
       "      <td>3.163710</td>\n",
       "      <td>0.010118</td>\n",
       "      <td>6751.0</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_49.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_49.4.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bcl6b</td>\n",
       "      <td>3.161393</td>\n",
       "      <td>0.010113</td>\n",
       "      <td>3317.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__CG15696_SOLEXA_FBgn0038833</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__CG15696_SOLEXA_FBgn0038833.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.156759</td>\n",
       "      <td>0.010103</td>\n",
       "      <td>18411.0</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.19</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.19.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Pax3, Barhl1</td>\n",
       "      <td>Isl2, Hoxc6, Hoxa6, Barhl2, Hoxa7, Hoxb6, Hoxb7</td>\n",
       "      <td>3.153118</td>\n",
       "      <td>0.010095</td>\n",
       "      <td>5506.0</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M02334</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M02334.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cdx2, Cdx1, Cdx4</td>\n",
       "      <td>3.143188</td>\n",
       "      <td>0.010073</td>\n",
       "      <td>724.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__LHX9_CYAATTAN_FL_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__LHX9_CYAATTAN_FL_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.141698</td>\n",
       "      <td>0.010069</td>\n",
       "      <td>23696.0</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M0456</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M0456.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.137064</td>\n",
       "      <td>0.010059</td>\n",
       "      <td>1034.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5152</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5152.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yy2, Yy1</td>\n",
       "      <td>3.134913</td>\n",
       "      <td>0.010054</td>\n",
       "      <td>886.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1212</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1212.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.126969</td>\n",
       "      <td>0.010037</td>\n",
       "      <td>6674.0</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0910.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0910.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hoxd8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.125810</td>\n",
       "      <td>0.010034</td>\n",
       "      <td>22432.0</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c2h2_zfs__M2013</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/c2h2_zfs__M2013.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Zfp352, Gm12845, Klf17</td>\n",
       "      <td>3.120845</td>\n",
       "      <td>0.010023</td>\n",
       "      <td>7279.0</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07780</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M07780.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.116873</td>\n",
       "      <td>0.010015</td>\n",
       "      <td>376.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_60.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_60.12.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.116542</td>\n",
       "      <td>0.010014</td>\n",
       "      <td>904.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_0.20</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_0.20.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Pax4, Uncx, Msx1, Lhx8, Hoxa2, Alx1, Gbx1, Lhx3, Pou1f1, Gbx2, En2</td>\n",
       "      <td>Pax4, Hoxd3, Lhx4, Alx3, Uncx, Hoxa5, Emx2, Hoxa4, Pdx1, Zfhx2, Hoxd8, Evx2, Meox2, Gm28308, Hesx1, Hoxa2, Prrxl1, Pax6, Lhx6, Pou6f1, Hoxa1, Nkx1-1, Hoxb1, Gbx2, Mixl1, Hoxb2, Emx1, Hoxd4, Hoxd1, Evx1, Vsx2, Noto, Gbx1, Vax2, En1, Vax1, Dlx3, Lbx2, Pou6f2, Lhx8, Mnx1, Vsx1, Alx1, Esx1, Meox1, Hoxb5, Lhx2, Alx4, Phox2a, Nkx1-2, En2</td>\n",
       "      <td>3.115384</td>\n",
       "      <td>0.010011</td>\n",
       "      <td>22321.0</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_26.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_26.16.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Hmx1, Msx3, Barhl2, Hmx2, Nkx2-5, Nkx6-1, Hmx3, Barhl1</td>\n",
       "      <td>Hmx1, Isl2, Nkx6-2, Uncx, Barhl2, Nkx1-1, Nkx2-5, Nkx6-1, Hmx3, Nkx2-3, Nkx1-2, Isl1, Barhl1</td>\n",
       "      <td>3.105950</td>\n",
       "      <td>0.009990</td>\n",
       "      <td>9569.0</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00389</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00389.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx</td>\n",
       "      <td>3.098834</td>\n",
       "      <td>0.009975</td>\n",
       "      <td>24249.0</td>\n",
       "      <td>123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6052</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M6052.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Meox2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.097675</td>\n",
       "      <td>0.009972</td>\n",
       "      <td>18980.0</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07439</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/transfac_pro__M07439.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Tef</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.095027</td>\n",
       "      <td>0.009966</td>\n",
       "      <td>23734.0</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00416</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00416.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>3.089731</td>\n",
       "      <td>0.009955</td>\n",
       "      <td>11081.0</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__H2.0_SOLEXA_FBgn0001170</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__H2.0_SOLEXA_FBgn0001170.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hlx</td>\n",
       "      <td>3.088242</td>\n",
       "      <td>0.009951</td>\n",
       "      <td>1141.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0358.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0358.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.083111</td>\n",
       "      <td>0.009940</td>\n",
       "      <td>1246.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Zen_SOLEXA_FBgn0004053</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Zen_SOLEXA_FBgn0004053.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.080132</td>\n",
       "      <td>0.009933</td>\n",
       "      <td>3514.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_26.15</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_26.15.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hlx, Dbx1, Dbx2</td>\n",
       "      <td>3.070533</td>\n",
       "      <td>0.009912</td>\n",
       "      <td>933.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5641</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5641.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Msx2</td>\n",
       "      <td>3.068381</td>\n",
       "      <td>0.009907</td>\n",
       "      <td>408.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__PHF8__ENCSR000AQH_1__m2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__PHF8__ENCSR000AQH_1__m2.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4921501E09Rik, Phf8</td>\n",
       "      <td>3.067885</td>\n",
       "      <td>0.009906</td>\n",
       "      <td>1206.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0628.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/jaspar__MA0628.1.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pou6f1</td>\n",
       "      <td>3.064906</td>\n",
       "      <td>0.009900</td>\n",
       "      <td>9773.0</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00328</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/tfdimers__MD00328.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nfyc, Nfyb, Nfya, Hnf1b, Hnf1a</td>\n",
       "      <td>3.064575</td>\n",
       "      <td>0.009899</td>\n",
       "      <td>15244.0</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_9.46</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_9.46.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.060106</td>\n",
       "      <td>0.009889</td>\n",
       "      <td>3259.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M1177</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M1177.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.057458</td>\n",
       "      <td>0.009883</td>\n",
       "      <td>12463.0</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>flyfactorsurvey__Scr_Cell_FBgn0003339</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/flyfactorsurvey__Scr_Cell_FBgn0003339.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxc5, Hoxa5, Hoxb5</td>\n",
       "      <td>3.056134</td>\n",
       "      <td>0.009880</td>\n",
       "      <td>13121.0</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_15.7</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_15.7.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Uncx, Hoxc6, Evx1, Dlx1, Hoxa3, Meox1, Hoxb3, Nkx6-1, Msx2</td>\n",
       "      <td>Lmx1a, Lmx1b, Lhx4, Uncx, Hoxa5, Emx2, Hoxa4, Hoxb6, Bsx, Pdx1, Otp, Lhx1, Isl2, Meox2, Nkx6-3, Hoxc6, Gm28308, Hoxa7, Hesx1, Hoxa2, Prrxl1, Rax, Lhx6, Hoxa1, Nkx1-1, Hoxb1, Isl1, Gsx2, Nkx6-2, Gbx2, Hoxb2, Hoxd4, Emx1, Hoxd1, Hoxb4, Noto, Vsx2, Gbx1, Hoxc4, Hoxc5, En1, Vax1, Hoxb7, Lhx9, Pou6f2, Prrx2, Lhx8, Lbx1, Hoxa6, Hoxb5, Vsx1, Meox1, Gsx1, Hoxb3, Lhx2, Lhx3, Nkx6-1, Lhx5, Prop1, Nkx1-2, En2</td>\n",
       "      <td>3.052990</td>\n",
       "      <td>0.009873</td>\n",
       "      <td>10404.0</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5396</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5396.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Eomes</td>\n",
       "      <td>3.047363</td>\n",
       "      <td>0.009861</td>\n",
       "      <td>22252.0</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>elemento__TAATTAA</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/elemento__TAATTAA.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.044218</td>\n",
       "      <td>0.009854</td>\n",
       "      <td>1484.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M5516</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/cisbp__M5516.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hlf</td>\n",
       "      <td>3.034288</td>\n",
       "      <td>0.009832</td>\n",
       "      <td>20755.0</td>\n",
       "      <td>114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_73.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/metacluster_73.12.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cux2</td>\n",
       "      <td>3.034123</td>\n",
       "      <td>0.009832</td>\n",
       "      <td>22818.0</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__HLF_full_NRTTACGTAAYN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__HLF_full_NRTTACGTAAYN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hlf</td>\n",
       "      <td>3.029654</td>\n",
       "      <td>0.009822</td>\n",
       "      <td>20752.0</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__SIX5__ENCSR000BJE_1__m3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__SIX5__ENCSR000BJE_1__m3.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Six5</td>\n",
       "      <td>3.023696</td>\n",
       "      <td>0.009809</td>\n",
       "      <td>5846.0</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__Meox2_DBD_NTAATKAN</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale__Meox2_DBD_NTAATKAN.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>Meox2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.021876</td>\n",
       "      <td>0.009805</td>\n",
       "      <td>12974.0</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__SREBF1_RTCAGGTGAY_eDBD_meth</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_cyt_meth__SREBF1_RTCAGGTGAY_eDBD_meth.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Srebf1</td>\n",
       "      <td>3.016911</td>\n",
       "      <td>0.009794</td>\n",
       "      <td>3563.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_tf_pairs__HOXA6_SYMATTAN_HT</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/taipale_tf_pairs__HOXA6_SYMATTAN_HT.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hoxa6</td>\n",
       "      <td>3.013270</td>\n",
       "      <td>0.009786</td>\n",
       "      <td>4986.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__PBX3__ENCSR000BGR_1__m2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v9nr_clust/logos/dbcorrdb__PBX3__ENCSR000BGR_1__m2.png\" width=\"200\" ></td>\n",
       "      <td>Top1K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pbx3</td>\n",
       "      <td>3.005160</td>\n",
       "      <td>0.009768</td>\n",
       "      <td>14807.0</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pycistarget.motif_enrichment_cistarget import *\n",
    "cistarget_results(cistarget_dict, name='Top1K')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc3ed2d2-c813-4f84-a0fc-76863d6f686d",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pycistarget.motif_enrichment_cistarget import *\n",
    "cistarget_results(cistarget_dict, name='Top1K')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "62bae4a8-acab-4c1d-aff1-59a5054450cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['Gsx2_(110r)', 'Nkx1-2_(110r)', 'Phox2a_(110r)', 'Gbx1_(136r)', 'En2_(116r)', 'Foxb1_(139r)', 'Meox1_(119r)', 'Hmx3_(53r)', 'Onecut2_(111r)', 'Eomes_(145r)', 'Nkx1-1_(110r)', 'Barx2_(110r)', 'Zscan10_(23r)', 'Foxa3_(139r)', 'Cebpg_(213r)', 'Hoxd1_(110r)', 'Lhx9_(110r)', 'Cebpb_(213r)', 'Foxc1_(139r)', 'Meox2_(101r)', 'Mga_(63r)', 'Isl1_(54r)', 'Hoxa1_(110r)', 'Evx2_(110r)', 'Arid3a_(32r)', 'Lhx4_(110r)', 'Hoxc6_(60r)', 'Pdx1_(157r)', 'Vsx1_(110r)', 'Msx2_(119r)', 'Foxl2_(139r)', 'Hoxb7_(110r)', 'Cebpd_(278r)', 'Tbx1_(122r)', 'Tcf7l2_(125r)', 'Lhx8_(136r)', 'Lhx2_(139r)', 'Sp4_(14r)', 'Dlx5_(110r)', 'Nrf1_(28r)', 'Prrx2_(110r)', 'Onecut1_(242r)', 'Cebpa_(284r)', 'Hnf4a_(142r)', 'Bsx_(110r)', 'Foxd1_(139r)', 'Hoxd8_(115r)', 'Hoxa5_(110r)', 'Barhl1_(71r)', 'Foxj3_(139r)', 'Foxl1_(139r)', 'Cebpe_(276r)', 'Hoxa2_(136r)', 'Cux1_(139r)', 'Tbx2_(145r)', 'Nanog_(104r)', 'Foxf2_(139r)', 'Alx3_(110r)', 'Nkx6-1_(127r)', 'Adnp2_(16r)', 'Hnf1b_(177r)', 'Mnx1_(110r)', 'Hoxb6_(110r)', 'Alx1_(151r)', 'Gbx2_(136r)', 'Uncx_(113r)', 'Lbx2_(110r)', 'Nobox_(111r)', 'Foxk1_(139r)', 'Barhl2_(53r)', 'Vax1_(110r)', 'En1_(110r)', 'Dlx3_(110r)', 'Pax7_(218r)', 'Hnf1a_(203r)', 'Tef_(161r)', 'Dbp_(263r)', 'Foxa1_(139r)', 'Arx_(139r)', 'Evx1_(119r)', 'Tbx3_(145r)', 'Lmx1a_(106r)', 'Lhx3_(124r)', 'Lhx6_(180r)', 'Pax4_(136r)', 'Pax6_(110r)', 'Rax_(110r)', 'Tbx20_(176r)', 'Hoxa6_(110r)', 'T_(145r)', 'Tbx4_(145r)', 'Barx1_(110r)', 'Isx_(110r)', 'Pou1f1_(105r)', 'Nkx2-5_(53r)', 'Msx3_(53r)', 'Alx4_(139r)', 'Shox2_(139r)', 'Tbx21_(176r)', 'Tbx6_(60r)', 'Prrx1_(181r)', 'Lmx1b_(106r)', 'Bbx_(139r)', 'Hoxc8_(110r)', 'Hoxb5_(110r)', 'Vsx2_(110r)', 'Tbx5_(145r)', 'Tbr1_(145r)', 'Hmx1_(53r)', 'Nfil3_(117r)', 'Hoxa7_(110r)', 'Dlx1_(60r)', 'Vax2_(110r)', 'Esx1_(110r)', 'Foxa2_(139r)', 'Hoxa3_(119r)', 'Pax3_(43r)', 'Hmx2_(53r)', 'Msx1_(105r)', 'Hlf_(263r)', 'Hoxd3_(176r)', 'Hoxb3_(119r)', 'Gsx2_extended_(146r)', 'Zfhx3_extended_(177r)', 'Nkx1-2_extended_(164r)', 'Phox2a_extended_(136r)', 'Zfp12_extended_(177r)', 'Dbx2_extended_(12r)', 'Egr1_extended_(52r)', 'Klf15_extended_(14r)', 'Tbx22_extended_(122r)', 'Gbx1_extended_(139r)', 'En2_extended_(197r)', 'Foxb1_extended_(243r)', 'Meox1_extended_(157r)', 'Hoxa4_extended_(150r)', 'Hmx3_extended_(53r)', 'Gm28308_extended_(157r)', 'Onecut2_extended_(232r)', 'Pou2f3_extended_(104r)', 'Foxk2_extended_(139r)', 'Eomes_extended_(188r)', 'Nkx1-1_extended_(164r)', 'Barx2_extended_(133r)', 'Hmbox1_extended_(123r)', 'Mybl2_extended_(261r)', 'Zscan10_extended_(23r)', 'E2f6_extended_(23r)', 'Ppargc1a_extended_(213r)', 'Foxa3_extended_(243r)', 'Dbx1_extended_(12r)', 'Cebpg_extended_(348r)', 'Hoxd1_extended_(139r)', 'Lhx9_extended_(196r)', 'Foxi2_extended_(139r)', 'Nfyb_extended_(134r)', 'Cebpb_extended_(361r)', 'Foxc1_extended_(243r)', 'Meox2_extended_(200r)', 'Foxp3_extended_(190r)', 'Foxj2_extended_(139r)', 'Cdx2_extended_(12r)', 'Foxp1_extended_(190r)', 'Hoxc4_extended_(121r)', 'Gm5294_extended_(139r)', 'Mga_extended_(217r)', 'Nfyc_extended_(85r)', 'Gm12845_extended_(46r)', 'Hnf4g_extended_(165r)', 'Isl1_extended_(155r)', 'Hoxa1_extended_(139r)', 'Hdac2_extended_(139r)', 'Evx2_extended_(213r)', 'Arid3a_extended_(32r)', 'Hsf1_extended_(213r)', 'Foxj1_extended_(139r)', 'Ddit3_extended_(185r)', 'Lhx4_extended_(157r)', 'Pou4f1_extended_(111r)', 'Hoxc6_extended_(118r)', 'Pdx1_extended_(172r)', 'Pou3f1_extended_(54r)', 'Vsx1_extended_(152r)', 'Msx2_extended_(120r)', 'Ep300_extended_(405r)', 'Bcl6b_extended_(28r)', 'Foxl2_extended_(139r)', 'Hoxb7_extended_(153r)', 'Cebpd_extended_(310r)', '4921501E09Rik_extended_(13r)', 'Tbx1_extended_(176r)', 'Taf1_extended_(13r)', 'Tcf7l2_extended_(243r)', 'Nkx2-3_extended_(53r)', 'Lhx8_extended_(145r)', 'Foxn3_extended_(139r)', 'Tlx2_extended_(135r)', 'Lhx2_extended_(212r)', 'Foxo1_extended_(232r)', 'Otp_extended_(114r)', 'Sp4_extended_(18r)', 'Dlx5_extended_(110r)', 'Crx_extended_(279r)', 'Cdx4_extended_(12r)', 'Nrf1_extended_(105r)', 'Tbx15_extended_(176r)', 'Prrx2_extended_(145r)', 'Pbx3_extended_(75r)', 'Pou2f1_extended_(105r)', 'Foxd2_extended_(243r)', 'Lhx1_extended_(122r)', 'Onecut1_extended_(275r)', 'Cebpa_extended_(356r)', 'Hnf4a_extended_(225r)', 'Bsx_extended_(141r)', 'Foxd1_extended_(139r)', 'Tbx10_extended_(176r)', 'Gata3_extended_(139r)', 'Hoxd8_extended_(184r)', 'Zfp352_extended_(46r)', 'Hoxa5_extended_(189r)', 'Barhl1_extended_(71r)', 'Foxj3_extended_(139r)', 'Foxl1_extended_(139r)', 'Cebpe_extended_(308r)', 'Hoxa2_extended_(157r)', 'Tlx1_extended_(104r)', 'Srebf1_extended_(26r)', 'Cux1_extended_(139r)', 'Six5_extended_(37r)', 'Tbx2_extended_(176r)', 'Cdx1_extended_(12r)', 'Nanog_extended_(104r)', 'Foxf2_extended_(139r)', 'Alx3_extended_(159r)', 'Klf17_extended_(46r)', 'Nkx6-1_extended_(156r)', 'Lhx5_extended_(122r)', 'Klf13_extended_(14r)', 'Hlx_extended_(18r)', 'Adnp2_extended_(16r)', 'Hnf1b_extended_(252r)', 'Noto_extended_(131r)', 'Mnx1_extended_(136r)', 'Hoxb6_extended_(153r)', 'Yy1_extended_(20r)', 'Alx1_extended_(151r)', 'Gbx2_extended_(152r)', 'Uncx_extended_(186r)', 'Lbx2_extended_(152r)', 'Nobox_extended_(111r)', 'Hoxb1_extended_(113r)', 'Foxk1_extended_(139r)', 'Foxi1_extended_(139r)', 'Hoxd4_extended_(150r)', 'Barhl2_extended_(71r)', 'Emx2_extended_(157r)', 'Foxi3_extended_(139r)', 'Vax1_extended_(139r)', 'En1_extended_(193r)', 'Dlx3_extended_(206r)', 'Sp1_extended_(74r)', 'Bmyc_extended_(185r)', 'Pax7_extended_(218r)', 'Myc_extended_(185r)', 'Hnf1a_extended_(245r)', 'Tcf12_extended_(139r)', 'Emx1_extended_(134r)', 'Tef_extended_(161r)', 'Mixl1_extended_(129r)', 'Sp3_extended_(18r)', 'Pou2f2 _extended_(104r)', 'Dbp_extended_(263r)', 'Foxa1_extended_(305r)', 'Arx_extended_(164r)', 'Evx1_extended_(213r)', 'Tbx3_extended_(252r)', 'Zfhx2_extended_(105r)', 'Lmx1a_extended_(144r)', 'Nkx6-3_extended_(112r)', 'Lhx3_extended_(143r)', 'Lhx6_extended_(197r)', 'Hoxb4_extended_(121r)', 'Foxp2_extended_(190r)', 'Tbx18_extended_(145r)', 'Pax4_extended_(226r)', 'Pax6_extended_(235r)', 'Foxc2_extended_(243r)', 'Pou6f2_extended_(113r)', 'Rax_extended_(228r)', 'Foxf1_extended_(139r)', 'Hoxc5_extended_(102r)', 'Prrxl1_extended_(115r)', 'Tbx20_extended_(176r)', 'Foxd3_extended_(243r)', 'Foxg1_extended_(139r)', 'Zbtb49_extended_(110r)', 'Foxn2_extended_(139r)', 'Hoxa6_extended_(157r)', 'T_extended_(176r)', 'Tbx4_extended_(145r)', 'Rreb1_extended_(16r)', 'Barx1_extended_(110r)', 'Isx_extended_(139r)', 'Nfya_extended_(85r)', 'Pou1f1_extended_(105r)', 'Hesx1_extended_(113r)', 'Lbx1_extended_(160r)', 'Nkx2-5_extended_(53r)', 'Hoxb2_extended_(231r)', 'Msx3_extended_(56r)', 'Alx4_extended_(259r)', 'Shox2_extended_(139r)', 'Tbx21_extended_(256r)', 'Tbx19_extended_(122r)', 'Tbx6_extended_(154r)', 'Cux2_extended_(189r)', 'Foxp4_extended_(190r)', 'Nfic_extended_(139r)', 'Yy2_extended_(10r)', 'Phf8_extended_(13r)', 'Prrx1_extended_(192r)', 'Foxq1_extended_(139r)', 'Lmx1b_extended_(148r)', 'Pou6f1_extended_(132r)', 'Sp2_extended_(74r)', 'Onecut3_extended_(188r)', 'Bbx_extended_(139r)', 'Tlx3_extended_(104r)', 'Hoxc8_extended_(110r)', 'Hoxb5_extended_(194r)', 'Gm43517_extended_(139r)', 'Gsx1_extended_(144r)', 'Vsx2_extended_(169r)', 'Skor1_extended_(110r)', 'Tbx5_extended_(145r)', 'Tbr1_extended_(176r)', 'Hmx1_extended_(53r)', 'Nfil3_extended_(263r)', 'Hoxa7_extended_(155r)', 'Foxb2_extended_(134r)', 'Dlx1_extended_(60r)', 'Foxr1_extended_(139r)', 'Vax2_extended_(150r)', 'Bcl3_extended_(139r)', 'Esx1_extended_(136r)', 'Foxa2_extended_(243r)', 'Hoxa3_extended_(119r)', 'Pax3_extended_(165r)', 'Hmx2_extended_(53r)', 'Msx1_extended_(129r)', 'Mbd4_extended_(139r)', 'Nkx6-2_extended_(138r)', 'Isl2_extended_(147r)', 'Foxm1_extended_(126r)', 'Prop1_extended_(60r)', 'Hlf_extended_(268r)', 'Hoxd3_extended_(181r)', 'Hoxb3_extended_(157r)'])"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cistarget_dict['Top1K'].cistromes['Region_set'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "585c5882-3058-4097-92ca-f8747b07610e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n"
     ]
    }
   ],
   "source": [
    "from pycistarget.utils import *\n",
    "cebpa_cistrome = cistarget_dict['Top1K'].cistromes['Region_set']['Tbx3_extended_(252r)']\n",
    "cebpa_cistrome_pr = pr.PyRanges(region_names_to_coordinates(cebpa_cistrome))\n",
    "cebpa_cistrome_pr.to_bed(path='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Liver_MO_v9_Tbx3_cistrome_1K.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "63c05a39-b082-4459-9209-f8d0f246e12e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-11 19:19:02,772 cisTarget    INFO     Reading cisTarget database\n",
      "2023-06-11 19:20:41,449 cisTarget    INFO     Running cisTarget for All_peaks which has 30356 regions\n",
      "2023-06-11 19:20:51,451 cisTarget    INFO     Annotating motifs for All_peaks\n",
      "2023-06-11 19:20:56,251 cisTarget    INFO     Getting cistromes for All_peaks\n",
      "2023-06-11 19:20:56,812 cisTarget    INFO     Running cisTarget for Top20K which has 26551 regions\n",
      "2023-06-11 19:21:06,370 cisTarget    INFO     Annotating motifs for Top20K\n",
      "2023-06-11 19:21:10,898 cisTarget    INFO     Getting cistromes for Top20K\n",
      "2023-06-11 19:21:11,405 cisTarget    INFO     Running cisTarget for Top10K which has 14314 regions\n",
      "2023-06-11 19:21:19,192 cisTarget    INFO     Annotating motifs for Top10K\n",
      "2023-06-11 19:21:22,813 cisTarget    INFO     Getting cistromes for Top10K\n",
      "2023-06-11 19:21:23,268 cisTarget    INFO     Running cisTarget for Top5K which has 7325 regions\n",
      "2023-06-11 19:21:30,033 cisTarget    INFO     Annotating motifs for Top5K\n",
      "2023-06-11 19:21:32,974 cisTarget    INFO     Getting cistromes for Top5K\n",
      "2023-06-11 19:21:33,290 cisTarget    INFO     Running cisTarget for Top1K which has 1531 regions\n",
      "2023-06-11 19:21:39,317 cisTarget    INFO     Annotating motifs for Top1K\n",
      "2023-06-11 19:21:41,776 cisTarget    INFO     Getting cistromes for Top1K\n",
      "2023-06-11 19:21:42,024 cisTarget    INFO     Running cisTarget for PP_ChIP_ov which has 362 regions\n",
      "2023-06-11 19:21:47,789 cisTarget    INFO     Annotating motifs for PP_ChIP_ov\n",
      "2023-06-11 19:21:50,136 cisTarget    INFO     Getting cistromes for PP_ChIP_ov\n",
      "2023-06-11 19:21:50,271 cisTarget    INFO     Running cisTarget for PP which has 924 regions\n",
      "2023-06-11 19:21:56,159 cisTarget    INFO     Annotating motifs for PP\n",
      "2023-06-11 19:21:58,553 cisTarget    INFO     Getting cistromes for PP\n",
      "2023-06-11 19:21:58,714 cisTarget    INFO     Running cisTarget for General which has 7092 regions\n",
      "2023-06-11 19:22:05,713 cisTarget    INFO     Annotating motifs for General\n",
      "2023-06-11 19:22:08,831 cisTarget    INFO     Getting cistromes for General\n",
      "2023-06-11 19:22:09,054 cisTarget    INFO     Running cisTarget for Pericentral which has 955 regions\n",
      "2023-06-11 19:22:14,832 cisTarget    INFO     Annotating motifs for Pericentral\n",
      "2023-06-11 19:22:17,233 cisTarget    INFO     Getting cistromes for Pericentral\n",
      "2023-06-11 19:22:17,398 cisTarget    INFO     Running cisTarget for Intermediate which has 2501 regions\n",
      "2023-06-11 19:22:24,059 cisTarget    INFO     Annotating motifs for Intermediate\n",
      "2023-06-11 19:22:26,728 cisTarget    INFO     Getting cistromes for Intermediate\n",
      "2023-06-11 19:22:26,952 cisTarget    INFO     Done!\n"
     ]
    }
   ],
   "source": [
    "# Run, using precomputed database\n",
    "from pycistarget.motif_enrichment_cistarget import *\n",
    "cistarget_dict = run_cistarget(ctx_db = '/staging/leuven/stg_00002/icistarget-data/make_rankings/v10_clust/CTX_mm10/CTX_mm10_SCREEN3_no_bg_with_mask/CTX_mm10_SCREEN3_no_bg_with_mask.regions_vs_motifs.rankings.v2.feather',\n",
    "                                                      region_sets = region_sets,\n",
    "                                                      specie = 'mus_musculus',\n",
    "                                                      auc_threshold = 0.005,\n",
    "                                                      nes_threshold = 3.0,\n",
    "                                                      rank_threshold = 0.05,\n",
    "                                                      annotation = ['Direct_annot', 'Orthology_annot'],\n",
    "                                                      annotation_version = 'v10nr_clust',\n",
    "                                                      path_to_motif_annotations = '/staging/leuven/stg_00002/lcb/cbravo/cluster_motif_collection_V10_no_desso_no_factorbook/snapshots/motifs-v10nr_clust-nr.mgi-m0.00001-o0.0.tbl',\n",
    "                                                      n_cpu = 1,\n",
    "                                                      _temp_dir='/scratch/leuven/313/vsc31305/ray_spill')\n",
    "# Save\n",
    "import pickle\n",
    "with open('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/cisTarget_dict_v10_nes3.0.pkl', 'wb') as f:\n",
    "  pickle.dump(cistarget_dict, f)\n",
    "\n",
    "\n",
    "# Run, using precomputed database\n",
    "#from pycistarget.motif_enrichment_cistarget import *\n",
    "#cistarget_dict = run_cistarget(ctx_db = '/staging/leuven/stg_00002/icistarget-data/make_rankings/v10_clust/CTX_mm10/CTX_mm10_SCREEN3_no_bg_with_mask/CTX_mm10_SCREEN3_no_bg_with_mask.regions_vs_motifs.rankings.v2.feather',\n",
    "#                                                      region_sets = region_sets,\n",
    "#                                                      specie = 'mus_musculus',\n",
    "#                                                      auc_threshold = 0.005,\n",
    "#                                                      nes_threshold = 0,\n",
    "#                                                      rank_threshold = 0.05,\n",
    "#                                                      annotation = ['Direct_annot', 'Orthology_annot'],\n",
    "#                                                      annotation_version = 'v10nr_clust',\n",
    "#                                                      path_to_motif_annotations = '/staging/leuven/stg_00002/lcb/cbravo/cluster_motif_collection_V10_no_desso_no_factorbook/snapshots/motifs-v10nr_clust-nr.mgi-m0.00001-o0.0.tbl',\n",
    "#                                                      n_cpu = 1,\n",
    "#                                                      _temp_dir='/scratch/leuven/313/vsc31305/ray_spill')\n",
    "# Save\n",
    "#import pickle\n",
    "#with open('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/cisTarget_dict_v10.pkl', 'wb') as f:\n",
    "#  pickle.dump(cistarget_dict, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b3c9a132-d049-4e93-954b-c1f4b9629af4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "infile = open('/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/cisTarget_dict_v10_nes3.0.pkl', 'rb')\n",
    "cistarget_dict = pickle.load(infile)\n",
    "infile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "7b119a37-a707-4463-943f-b097090ccce8",
   "metadata": {},
   "outputs": [],
   "source": [
    "for x in cistarget_dict.keys():\n",
    "    out_file = '/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/ctx_SCREEN_db/'+ str(x) + '.html'\n",
    "    cistarget_dict[x].motif_enrichment.to_html(open(out_file, 'w'), escape=False, col_space=80)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "7a07c008-56d9-45b4-ab87-e4dc31efe3fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['Foxl2_(199r)', 'Tbx20_(173r)', 'Cebpd_(251r)', 'Foxd3_(103r)', 'Hnf1b_(332r)', 'Tbx4_(173r)', 'E2f2_(126r)', 'T_(155r)', 'Klf10_(87r)', 'Tlx2_(129r)', 'Nfe2l1_(56r)', 'Ppara_(185r)', 'Nr2f2_(185r)', 'Klf7_(87r)', 'Nfya_(96r)', 'Foxb1_(199r)', 'Pbx1_(77r)', 'Sp4_(87r)', 'E2f1_(179r)', 'Nr2f6_(171r)', 'Eomes_(173r)', 'Nrf1_(56r)', 'Tbx21_(173r)', 'Foxk1_(149r)', 'Tbx6_(214r)', 'Zscan10_(56r)', 'Pbx3_(77r)', 'E2f6_(126r)', 'Klf2_(87r)', 'Foxa3_(279r)', 'Tfdp1_(126r)', 'Sp1_(144r)', 'E2f5_(126r)', 'Sp6_(87r)', 'Nr1h3_(185r)', 'Cebpg_(301r)', 'Hnf1a_(285r)', 'Foxd2_(103r)', 'Hnf4a_(325r)', 'Cebpa_(251r)', 'Nfyb_(96r)', 'E2f7_(126r)', 'Cebpb_(279r)', 'Sp3_(144r)', 'Foxa1_(237r)', 'Tbx5_(173r)', 'E2f3_(221r)', 'Nr2c2_(185r)', 'Ppard_(185r)', 'Klf14_(87r)', 'Tbr1_(173r)', 'Tbx3_(230r)', 'Sp7_(87r)', 'Foxp3_(155r)', 'Foxj3_(149r)', 'Rxra_(185r)', 'Sp5_(117r)', 'Foxl1_(149r)', 'Cebpe_(325r)', 'Mga_(173r)', 'Foxa2_(237r)', 'E2f4_(179r)', 'Tbx2_(173r)', 'Foxf1_(149r)', 'Klf17_(87r)', 'Nfyc_(96r)', 'Hlf_(251r)', 'Klf13_(87r)', 'Zfhx3_extended_(157r)', 'Hoxa13_extended_(143r)', 'Arid5b_extended_(180r)', 'Zbtb2_extended_(56r)', 'Tbp_extended_(52r)', 'Fos_extended_(77r)', 'Egr1_extended_(56r)', 'Klf15_extended_(87r)', 'Klf10_extended_(87r)', 'Tbx22_extended_(173r)', 'Ppara_extended_(185r)', 'Nr2f2_extended_(237r)', 'Tfdp2_extended_(126r)', 'Foxb1_extended_(254r)', 'Onecut2_extended_(221r)', 'Sp8_extended_(87r)', 'Nr2f6_extended_(261r)', 'Eomes_extended_(173r)', 'Hmbox1_extended_(197r)', 'Zscan10_extended_(56r)', 'E2f6_extended_(165r)', 'Irf3_extended_(77r)', 'Foxa3_extended_(349r)', 'Nr1h3_extended_(185r)', 'Foxo3_extended_(180r)', 'Cebpg_extended_(301r)', 'Zfy2_extended_(201r)', 'Nfyb_extended_(218r)', 'Gcm1_extended_(33r)', 'Cebpb_extended_(292r)', 'Foxc1_extended_(289r)', 'Foxp3_extended_(155r)', 'Foxj2_extended_(149r)', 'Pparg_extended_(185r)', 'Rxra_extended_(185r)', 'Cdx2_extended_(142r)', 'Foxp1_extended_(221r)', 'Mga_extended_(214r)', 'Ikzf5_extended_(149r)', 'Taf6_extended_(52r)', 'Lcorl_extended_(185r)', 'Hnf4g_extended_(254r)', 'Hdac2_extended_(199r)', 'Hsf1_extended_(162r)', 'Pdx1_extended_(52r)', 'Nfyc_extended_(218r)', 'Ep300_extended_(251r)', 'Foxl2_extended_(199r)', 'Cebpd_extended_(251r)', 'Hes2_extended_(251r)', 'Tbx1_extended_(214r)', 'Tlx2_extended_(138r)', 'Klf7_extended_(87r)', 'Paxip1_extended_(185r)', 'Nr2e3_extended_(171r)', 'Sp4_extended_(144r)', 'Crx_extended_(299r)', 'Nrf1_extended_(56r)', 'Tbx15_extended_(214r)', 'Dlx6_extended_(149r)', 'Pbx3_extended_(90r)', 'E2f5_extended_(126r)', 'Foxd2_extended_(289r)', 'E2f7_extended_(126r)', 'Hnf4a_extended_(325r)', 'Cebpa_extended_(251r)', 'Onecut1_extended_(197r)', 'Foxj3_extended_(221r)', 'Foxl1_extended_(149r)', 'Cebpe_extended_(325r)', 'Hmg20b_extended_(149r)', 'Klf8_extended_(87r)', 'Cux1_extended_(177r)', 'Tbx2_extended_(173r)', 'Sp9_extended_(87r)', 'Foxf2_extended_(199r)', 'Drap1_extended_(77r)', 'Klf12_extended_(87r)', 'Klf17_extended_(87r)', 'Klf13_extended_(87r)', 'Hnf1b_extended_(407r)', 'Ahdc1_extended_(180r)', 'E2f2_extended_(126r)', 'Hmgxb3_extended_(77r)', 'Zfp217_extended_(185r)', 'Mbd2_extended_(76r)', 'Pbx1_extended_(77r)', 'Runx1_extended_(40r)', 'Maz_extended_(87r)', 'Foxk1_extended_(237r)', 'Foxi1_extended_(259r)', 'Klf11_extended_(87r)', 'Cebpz_extended_(77r)', 'Tfdp1_extended_(126r)', 'Sp1_extended_(144r)', 'Klf1_extended_(170r)', 'Zfp644_extended_(201r)', 'Hnf1a_extended_(372r)', 'Tcf12_extended_(199r)', 'Tef_extended_(158r)', 'Mixl1_extended_(201r)', 'Sp3_extended_(144r)', 'Dbp_extended_(251r)', 'Foxa1_extended_(317r)', 'E2f3_extended_(221r)', 'Nr2c2_extended_(237r)', 'Klf14_extended_(87r)', 'Tbx3_extended_(333r)', 'Klf5_extended_(87r)', 'Sp7_extended_(87r)', 'Nr2f1_extended_(237r)', 'Sp5_extended_(117r)', 'Tbx18_extended_(214r)', 'E2f4_extended_(179r)', 'Foxc2_extended_(289r)', 'Pax4_extended_(107r)', 'Rax_extended_(159r)', 'Foxf1_extended_(149r)', 'Ski_extended_(149r)', 'Tbx20_extended_(214r)', 'Foxd3_extended_(327r)', 'Tbx4_extended_(173r)', 'T_extended_(214r)', 'Nfe2l1_extended_(56r)', 'Nfya_extended_(218r)', 'Pou1f1_extended_(41r)', 'Rb1_extended_(126r)', 'Patz1_extended_(87r)', 'E2f1_extended_(179r)', 'Hoxb2_extended_(147r)', 'Alx4_extended_(160r)', 'Tbx21_extended_(326r)', 'Tbx19_extended_(214r)', 'Tbx6_extended_(214r)', 'Cux2_extended_(150r)', 'Klf2_extended_(87r)', 'Foxp4_extended_(180r)', 'Foxq1_extended_(199r)', 'Hoxa10_extended_(92r)', 'Sp2_extended_(186r)', 'Sp6_extended_(87r)', 'Onecut3_extended_(150r)', 'Zfp281_extended_(87r)', 'Fli1_extended_(32r)', 'Tbx5_extended_(173r)', 'Rxrg_extended_(254r)', 'Ppard_extended_(185r)', 'Tbr1_extended_(173r)', 'Foxa2_extended_(317r)', 'Zfy1_extended_(201r)', 'Myb_extended_(117r)', 'Nkx6-2_extended_(143r)', 'Lcor_extended_(64r)', 'Foxm1_extended_(286r)', 'Gatad2a_extended_(251r)', 'Hlf_extended_(307r)', 'Foxe1_extended_(149r)'])"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cistarget_dict['Top1K'].cistromes['Region_set'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "71790c75-d3c1-47c3-8c56-a2ee69c20729",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n"
     ]
    }
   ],
   "source": [
    "from pycistarget.utils import *\n",
    "cebpa_cistrome = cistarget_dict['Top1K'].cistromes['Region_set']['Tbx3_extended_(333r)']\n",
    "cebpa_cistrome_pr = pr.PyRanges(region_names_to_coordinates(cebpa_cistrome))\n",
    "cebpa_cistrome_pr.to_bed(path='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/SCREEN_v10_Tbx3_cistrome_1K.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "4fd58ca2-b473-47d8-8b29-2e4878d72515",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th style=\"min-width: 80px;\"></th>\n",
       "      <th style=\"min-width: 80px;\">Logo</th>\n",
       "      <th style=\"min-width: 80px;\">Region_set</th>\n",
       "      <th style=\"min-width: 80px;\">Direct_annot</th>\n",
       "      <th style=\"min-width: 80px;\">Orthology_annot</th>\n",
       "      <th style=\"min-width: 80px;\">NES</th>\n",
       "      <th style=\"min-width: 80px;\">AUC</th>\n",
       "      <th style=\"min-width: 80px;\">Rank_at_max</th>\n",
       "      <th style=\"min-width: 80px;\">Motif_hits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>metacluster_46.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_46.4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Cebpe, Cebpd, Cebpg, Cebpa, Cebpb, Hlf</td>\n",
       "      <td>Cebpe, Cebpd, Cebpg, Hes2, Cebpa, Cebpb, Dbp, Gatad2a, Ep300</td>\n",
       "      <td>7.226552</td>\n",
       "      <td>0.018842</td>\n",
       "      <td>55015.0</td>\n",
       "      <td>1144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_174.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_174.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nfya, Pbx1, Nfyb, Pbx3, Nfyc</td>\n",
       "      <td>Nfya, Nfyb, Fos, Foxi1, Drap1, Irf3, Pbx3, Hmgxb3, Nfyc, Cebpz, Sp2</td>\n",
       "      <td>6.657431</td>\n",
       "      <td>0.017666</td>\n",
       "      <td>8337.0</td>\n",
       "      <td>229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_174.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_174.3.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mbd2, Nfya, Nfyb, Pbx3, Nfyc</td>\n",
       "      <td>6.018766</td>\n",
       "      <td>0.016346</td>\n",
       "      <td>7081.0</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_39.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_39.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf4a</td>\n",
       "      <td>Zfy2, Hnf4a, Mixl1, Zfy1, Hnf4g, Nr2f6, Rxrg, Zfp644</td>\n",
       "      <td>5.930443</td>\n",
       "      <td>0.016163</td>\n",
       "      <td>51741.0</td>\n",
       "      <td>938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.13</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.13.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>5.733564</td>\n",
       "      <td>0.015756</td>\n",
       "      <td>29356.0</td>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.3.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>5.730934</td>\n",
       "      <td>0.015751</td>\n",
       "      <td>40286.0</td>\n",
       "      <td>661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.10</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.10.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>5.677605</td>\n",
       "      <td>0.015640</td>\n",
       "      <td>52026.0</td>\n",
       "      <td>751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.12</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.12.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>5.676597</td>\n",
       "      <td>0.015638</td>\n",
       "      <td>48751.0</td>\n",
       "      <td>740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.9</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.9.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>5.552926</td>\n",
       "      <td>0.015383</td>\n",
       "      <td>48806.0</td>\n",
       "      <td>769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.8</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.8.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>5.533610</td>\n",
       "      <td>0.015343</td>\n",
       "      <td>28218.0</td>\n",
       "      <td>526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M01815</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/cisbp__M01815.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Cebpe</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.522630</td>\n",
       "      <td>0.015320</td>\n",
       "      <td>55516.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.11</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.11.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.508470</td>\n",
       "      <td>0.015291</td>\n",
       "      <td>47787.0</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.7</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.7.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>5.501809</td>\n",
       "      <td>0.015277</td>\n",
       "      <td>28874.0</td>\n",
       "      <td>503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.21</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.21.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>5.492831</td>\n",
       "      <td>0.015258</td>\n",
       "      <td>32842.0</td>\n",
       "      <td>526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_120.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_120.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nr1h3, Hnf4a, Rxra, Nr2c2, Ppard, Ppara, Nr2f2</td>\n",
       "      <td>Paxip1, Hnf4a, Lcorl, Hnf4g, Rxrg, Nr2c2, Zfp217, Pparg, Nr2f1</td>\n",
       "      <td>5.409206</td>\n",
       "      <td>0.015086</td>\n",
       "      <td>51179.0</td>\n",
       "      <td>890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_174.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_174.6.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nfya, Nfyb, Nfyc</td>\n",
       "      <td>Nfyb</td>\n",
       "      <td>5.222797</td>\n",
       "      <td>0.014700</td>\n",
       "      <td>17459.0</td>\n",
       "      <td>316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.15</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.15.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>5.056830</td>\n",
       "      <td>0.014357</td>\n",
       "      <td>53360.0</td>\n",
       "      <td>794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbtfbs__E2F8_representative_N1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/dbtfbs__E2F8_representative_N1.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E2f8</td>\n",
       "      <td>4.991029</td>\n",
       "      <td>0.014221</td>\n",
       "      <td>55518.0</td>\n",
       "      <td>845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.17</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.17.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.984551</td>\n",
       "      <td>0.014208</td>\n",
       "      <td>27371.0</td>\n",
       "      <td>472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_170.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_170.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Sp5, Sp4, Klf13, E2f4, E2f1, Sp3, Klf14, Klf2, Sp7, Klf10, Klf17, Sp1, Klf7, Sp6</td>\n",
       "      <td>Klf15, Klf10, Klf7, Patz1, Sp4, Maz, Sp8, Klf11, Klf2, Sp1, Klf1, Sp2, Sp6, Zfp281, Klf14, Klf5, Sp7, Sp5, Klf8, Sp9, Klf12, Klf17</td>\n",
       "      <td>4.967486</td>\n",
       "      <td>0.014173</td>\n",
       "      <td>52959.0</td>\n",
       "      <td>813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>homer__ATTGCGCAAC_CEBP</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/homer__ATTGCGCAAC_CEBP.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Cebpb</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.907430</td>\n",
       "      <td>0.014048</td>\n",
       "      <td>53821.0</td>\n",
       "      <td>956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_3.11</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_3.11.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Sp3, Sp1</td>\n",
       "      <td>Sp4, Sp3, Sp1, Sp2</td>\n",
       "      <td>4.893061</td>\n",
       "      <td>0.014019</td>\n",
       "      <td>50337.0</td>\n",
       "      <td>787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.5</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.5.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.889200</td>\n",
       "      <td>0.014011</td>\n",
       "      <td>34331.0</td>\n",
       "      <td>563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M04761</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M04761.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hsf1</td>\n",
       "      <td>4.815011</td>\n",
       "      <td>0.013857</td>\n",
       "      <td>55150.0</td>\n",
       "      <td>753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M01819</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/cisbp__M01819.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nfil3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.787777</td>\n",
       "      <td>0.013801</td>\n",
       "      <td>55322.0</td>\n",
       "      <td>772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.20</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.20.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>4.773957</td>\n",
       "      <td>0.013772</td>\n",
       "      <td>39184.0</td>\n",
       "      <td>605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00002</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/tfdimers__MD00002.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Crx</td>\n",
       "      <td>4.708470</td>\n",
       "      <td>0.013637</td>\n",
       "      <td>5951.0</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_55.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_55.3.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nfe2l1, Zscan10, Nrf1</td>\n",
       "      <td>Egr1, Zbtb2, E2f6, Nrf1</td>\n",
       "      <td>4.664263</td>\n",
       "      <td>0.013546</td>\n",
       "      <td>51854.0</td>\n",
       "      <td>706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_3.8</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_3.8.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Sp5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.637422</td>\n",
       "      <td>0.013490</td>\n",
       "      <td>54498.0</td>\n",
       "      <td>784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>kznf__KLF1_Barrera2016.2_PBM</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/kznf__KLF1_Barrera2016.2_PBM.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Klf1</td>\n",
       "      <td>4.602977</td>\n",
       "      <td>0.013419</td>\n",
       "      <td>55401.0</td>\n",
       "      <td>966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.19</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.19.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.562918</td>\n",
       "      <td>0.013336</td>\n",
       "      <td>37933.0</td>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>4.517271</td>\n",
       "      <td>0.013242</td>\n",
       "      <td>43041.0</td>\n",
       "      <td>679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.14</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.14.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.471337</td>\n",
       "      <td>0.013147</td>\n",
       "      <td>54431.0</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.18</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.18.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>4.465042</td>\n",
       "      <td>0.013134</td>\n",
       "      <td>38392.0</td>\n",
       "      <td>625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_195.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_195.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Sp4, Sp3, Egr1, Klf7</td>\n",
       "      <td>Klf4, Klf6, Klf15, Klf10, Klf7, Sp8, Sp4, Klf11, Klf2, Klf9, Sp1, Klf1, Sp6, Sp3, Klf16, Klf14, Zfp352, Klf3, Klf5, Egr2, Sp7, Sp5, Klf8, Gm12845, Sp9, Klf12, Klf17, Klf13</td>\n",
       "      <td>4.458970</td>\n",
       "      <td>0.013121</td>\n",
       "      <td>52299.0</td>\n",
       "      <td>707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_41.36</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_41.36.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1a, Hnf1b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.439496</td>\n",
       "      <td>0.013081</td>\n",
       "      <td>38255.0</td>\n",
       "      <td>595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.16</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.16.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.416319</td>\n",
       "      <td>0.013033</td>\n",
       "      <td>34385.0</td>\n",
       "      <td>568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hdpi__ELF2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/hdpi__ELF2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Elf2</td>\n",
       "      <td>4.343020</td>\n",
       "      <td>0.012882</td>\n",
       "      <td>47273.0</td>\n",
       "      <td>676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hdpi__ETV4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/hdpi__ETV4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Etv4</td>\n",
       "      <td>4.320982</td>\n",
       "      <td>0.012836</td>\n",
       "      <td>45299.0</td>\n",
       "      <td>649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>kznf__ZBTB14_Schmitges2016_RCADE</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/kznf__ZBTB14_Schmitges2016_RCADE.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Zbtb14</td>\n",
       "      <td>4.263203</td>\n",
       "      <td>0.012717</td>\n",
       "      <td>54892.0</td>\n",
       "      <td>962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hdpi__NRL</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/hdpi__NRL.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nrl</td>\n",
       "      <td>4.211196</td>\n",
       "      <td>0.012609</td>\n",
       "      <td>55082.0</td>\n",
       "      <td>786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_165.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_165.4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut3, Cux1, Onecut1, Onecut2, Cux2</td>\n",
       "      <td>4.200138</td>\n",
       "      <td>0.012586</td>\n",
       "      <td>55499.0</td>\n",
       "      <td>827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_39.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_39.3.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nr2f6, Hnf4a</td>\n",
       "      <td>Nr2c2, Nr2e3, Nr2f2, Nr2f1</td>\n",
       "      <td>4.160851</td>\n",
       "      <td>0.012505</td>\n",
       "      <td>54705.0</td>\n",
       "      <td>872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_129.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_129.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Foxd3, Foxd2</td>\n",
       "      <td>Foxb1, Foxd2, Foxa2, Foxc1, Foxa1, Foxc2, Foxa3</td>\n",
       "      <td>4.114183</td>\n",
       "      <td>0.012409</td>\n",
       "      <td>19581.0</td>\n",
       "      <td>324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_3.7</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_3.7.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Sp3, Sp1</td>\n",
       "      <td>Sp5, Sp4, Sp3, Sp1, Sp2</td>\n",
       "      <td>4.098715</td>\n",
       "      <td>0.012377</td>\n",
       "      <td>55319.0</td>\n",
       "      <td>825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M00803</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M00803.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>E2f3, Tfdp1, E2f1, E2f4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.032600</td>\n",
       "      <td>0.012240</td>\n",
       "      <td>53260.0</td>\n",
       "      <td>853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M03862</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M03862.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Ybx1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.014317</td>\n",
       "      <td>0.012202</td>\n",
       "      <td>3180.0</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00041</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/tfdimers__MD00041.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hmbox1, Crx</td>\n",
       "      <td>3.935587</td>\n",
       "      <td>0.012039</td>\n",
       "      <td>12544.0</td>\n",
       "      <td>248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__E2F2_GCGCGCGCGYW_eDBD_repr</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/taipale_cyt_meth__E2F2_GCGCGCGCGYW_eDBD_repr.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E2f2</td>\n",
       "      <td>3.935116</td>\n",
       "      <td>0.012039</td>\n",
       "      <td>39113.0</td>\n",
       "      <td>800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M02089</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M02089.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>E2f3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.888762</td>\n",
       "      <td>0.011943</td>\n",
       "      <td>55514.0</td>\n",
       "      <td>885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_127.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_127.1.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Foxb1, Foxl2, Foxa2, Foxa1, Foxa3</td>\n",
       "      <td>Foxq1, Foxb1, Foxm1, Foxl2, Foxd2, Foxa2, Foxd3, Tcf12, Foxc1, Foxa1, Foxk1, Foxf2, Foxc2, Foxi1, Hdac2, Foxa3</td>\n",
       "      <td>3.882494</td>\n",
       "      <td>0.011930</td>\n",
       "      <td>55434.0</td>\n",
       "      <td>861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.23</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.23.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.851203</td>\n",
       "      <td>0.011865</td>\n",
       "      <td>55460.0</td>\n",
       "      <td>747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_134.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_134.4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxo3, Foxa2, Arid5b, Ahdc1, Foxa1, Foxp4, Foxj3, Foxa3, Foxp1</td>\n",
       "      <td>3.832659</td>\n",
       "      <td>0.011827</td>\n",
       "      <td>55350.0</td>\n",
       "      <td>821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jaspar__MA0502.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/jaspar__MA0502.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nfyb</td>\n",
       "      <td>3.807964</td>\n",
       "      <td>0.011776</td>\n",
       "      <td>10915.0</td>\n",
       "      <td>202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.22</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.22.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>3.763913</td>\n",
       "      <td>0.011685</td>\n",
       "      <td>54327.0</td>\n",
       "      <td>737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01869</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M01869.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Cebpg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.660436</td>\n",
       "      <td>0.011471</td>\n",
       "      <td>55483.0</td>\n",
       "      <td>800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_156.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_156.3.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Tef, Hlf, Dbp, Nfil3</td>\n",
       "      <td>Tef, Dbp, Gm4125, Nfil3, Hlf</td>\n",
       "      <td>3.655554</td>\n",
       "      <td>0.011461</td>\n",
       "      <td>54541.0</td>\n",
       "      <td>802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_public__M00040</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_public__M00040.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Atf2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.596964</td>\n",
       "      <td>0.011340</td>\n",
       "      <td>12539.0</td>\n",
       "      <td>238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M04817</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M04817.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Zbtb33</td>\n",
       "      <td>3.596742</td>\n",
       "      <td>0.011339</td>\n",
       "      <td>54505.0</td>\n",
       "      <td>804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M02266</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M02266.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hnf1b</td>\n",
       "      <td>3.553595</td>\n",
       "      <td>0.011250</td>\n",
       "      <td>32307.0</td>\n",
       "      <td>480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.24</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.24.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.544486</td>\n",
       "      <td>0.011231</td>\n",
       "      <td>43277.0</td>\n",
       "      <td>633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M08883</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M08883.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxm1</td>\n",
       "      <td>3.535692</td>\n",
       "      <td>0.011213</td>\n",
       "      <td>55384.0</td>\n",
       "      <td>768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>swissregulon__hs__GMEB2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/swissregulon__hs__GMEB2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gmeb2</td>\n",
       "      <td>3.508039</td>\n",
       "      <td>0.011156</td>\n",
       "      <td>52750.0</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>swissregulon__hs__ONECUT1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/swissregulon__hs__ONECUT1.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut1</td>\n",
       "      <td>3.479667</td>\n",
       "      <td>0.011097</td>\n",
       "      <td>55513.0</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>homer__TATTGAYY_Hnf6b</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/homer__TATTGAYY_Hnf6b.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Onecut2</td>\n",
       "      <td>3.412152</td>\n",
       "      <td>0.010958</td>\n",
       "      <td>51653.0</td>\n",
       "      <td>729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_34.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_34.4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Zfp143, Thap11, Tbx2</td>\n",
       "      <td>Zfp143, Thap11, Smarcc2, Six5, Hcfc1, Ets1</td>\n",
       "      <td>3.398398</td>\n",
       "      <td>0.010929</td>\n",
       "      <td>1404.0</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_74.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_74.6.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Hnf1a</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.397390</td>\n",
       "      <td>0.010927</td>\n",
       "      <td>52795.0</td>\n",
       "      <td>702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M01861</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M01861.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Atf1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.363888</td>\n",
       "      <td>0.010858</td>\n",
       "      <td>40333.0</td>\n",
       "      <td>560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_61.1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_61.1.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Deaf1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.352109</td>\n",
       "      <td>0.010833</td>\n",
       "      <td>51111.0</td>\n",
       "      <td>679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_34.3</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_34.3.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Setdb1, Zfp143, Six5, Zfp523, Egr1, Ets1</td>\n",
       "      <td>3.342621</td>\n",
       "      <td>0.010814</td>\n",
       "      <td>1156.0</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_67.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_67.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>E2f3, Foxn4, E2f2</td>\n",
       "      <td>Acaa1a, Acaa1b</td>\n",
       "      <td>3.340397</td>\n",
       "      <td>0.010809</td>\n",
       "      <td>54648.0</td>\n",
       "      <td>859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M04866</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M04866.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hdac2</td>\n",
       "      <td>3.337845</td>\n",
       "      <td>0.010804</td>\n",
       "      <td>55176.0</td>\n",
       "      <td>939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M07938</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/cisbp__M07938.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E2f1</td>\n",
       "      <td>3.305664</td>\n",
       "      <td>0.010737</td>\n",
       "      <td>55393.0</td>\n",
       "      <td>842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_33.8</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_33.8.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Zbtb33, Chd2</td>\n",
       "      <td>Brca1, Chd2, Nr3c1, Zbtb33, Ets1</td>\n",
       "      <td>3.283207</td>\n",
       "      <td>0.010691</td>\n",
       "      <td>55472.0</td>\n",
       "      <td>673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>swissregulon__mm__Cebpe</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/swissregulon__mm__Cebpe.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Cebpe</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.267477</td>\n",
       "      <td>0.010658</td>\n",
       "      <td>49691.0</td>\n",
       "      <td>755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__FOXD3_DBD_RTAAAYA</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/taipale__FOXD3_DBD_RTAAAYA.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Foxd3</td>\n",
       "      <td>3.214802</td>\n",
       "      <td>0.010550</td>\n",
       "      <td>51389.0</td>\n",
       "      <td>742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbtfbs__HLF_HepG2_ENCSR528PSI_merged_N1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/dbtfbs__HLF_HepG2_ENCSR528PSI_merged_N1.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hlf</td>\n",
       "      <td>3.200498</td>\n",
       "      <td>0.010520</td>\n",
       "      <td>54116.0</td>\n",
       "      <td>728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M12628</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M12628.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lcor</td>\n",
       "      <td>3.163161</td>\n",
       "      <td>0.010443</td>\n",
       "      <td>55101.0</td>\n",
       "      <td>721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_11.2</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_11.2.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nr2f2, Rara, Hnf4a, Nr4a1, Nr0b1, Rarg, Nr4a2, Ppara, Rxra, Nr2f1, Esrra</td>\n",
       "      <td>Nr5a2, Rara, Ppara, Nr5a1, Nr4a1, Nr2f1, Nr2f6, Rarg, Nr4a2, Nr2c1, Nr4a3, Rxrb, Rxra, Nr2f2, Esrra</td>\n",
       "      <td>3.149904</td>\n",
       "      <td>0.010415</td>\n",
       "      <td>52651.0</td>\n",
       "      <td>713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_188.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_188.6.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Foxa2, Foxa1, Foxk1, Foxf1, Foxj3, Foxa3, Foxl1</td>\n",
       "      <td>Hmg20b, Foxd3, Foxd2, Ikzf5, Foxa2, Foxe1, Foxc1, Dlx6, Foxk1, Foxc2, Foxj2, Foxa3, Foxm1, Ski, Foxp1, Foxl1</td>\n",
       "      <td>3.122841</td>\n",
       "      <td>0.010359</td>\n",
       "      <td>55452.0</td>\n",
       "      <td>813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbtfbs__GMEB1_K562_ENCSR928KOR_merged_N1</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/dbtfbs__GMEB1_K562_ENCSR928KOR_merged_N1.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gmeb1</td>\n",
       "      <td>3.122003</td>\n",
       "      <td>0.010358</td>\n",
       "      <td>48882.0</td>\n",
       "      <td>622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_3.4</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_3.4.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sp3, Sp1, Sp2</td>\n",
       "      <td>3.119124</td>\n",
       "      <td>0.010352</td>\n",
       "      <td>55307.0</td>\n",
       "      <td>755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M00652</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_pro__M00652.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nrf1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.096588</td>\n",
       "      <td>0.010305</td>\n",
       "      <td>23167.0</td>\n",
       "      <td>370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_165.5</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_165.5.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Onecut2, Onecut1</td>\n",
       "      <td>Onecut3, Cux1, Onecut2, Onecut1, Cux2</td>\n",
       "      <td>3.096091</td>\n",
       "      <td>0.010304</td>\n",
       "      <td>53306.0</td>\n",
       "      <td>730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_46.5</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_46.5.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tef, Hlf</td>\n",
       "      <td>3.091615</td>\n",
       "      <td>0.010295</td>\n",
       "      <td>53967.0</td>\n",
       "      <td>740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metacluster_111.6</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/metacluster_111.6.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gmeb1, Gmeb2</td>\n",
       "      <td>3.054514</td>\n",
       "      <td>0.010218</td>\n",
       "      <td>42360.0</td>\n",
       "      <td>576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_public__M00209</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/transfac_public__M00209.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>Nfya, Nfyb</td>\n",
       "      <td>Nfyc</td>\n",
       "      <td>3.034596</td>\n",
       "      <td>0.010177</td>\n",
       "      <td>5235.0</td>\n",
       "      <td>114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tfdimers__MD00200</th>\n",
       "      <td><img src=\"https://motifcollections.aertslab.org/v10nr_clust/logos/tfdimers__MD00200.png\" width=\"200\" ></td>\n",
       "      <td>Top5K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pou1f1, Tlx2</td>\n",
       "      <td>3.027621</td>\n",
       "      <td>0.010163</td>\n",
       "      <td>7384.0</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pycistarget.motif_enrichment_cistarget import *\n",
    "cistarget_results(cistarget_dict, name='Top5K')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bef66ce9-4dd9-4a09-b5b2-2d4f9c89351a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-05-22 20:54:51,178 cisTopic     INFO     Extending and merging peaks per class\n",
      "2023-05-22 20:55:13,289 cisTopic     INFO     Normalizing peak scores\n",
      "2023-05-22 20:55:13,411 cisTopic     INFO     Merging peaks\n",
      "Warning! Start and End columns now have different dtypes: int64 and int32\n",
      "2023-05-22 20:55:46,246 cisTopic     INFO     Done!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n"
     ]
    }
   ],
   "source": [
    "# Generate ChIP-seq consensus peaks\n",
    "import os\n",
    "import pyranges as pr\n",
    "import pandas as pd\n",
    "\n",
    "path='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/MACS/'\n",
    "files = [path+'Hnf4a_ERR235763'+'_peaks.narrowPeak',\n",
    "path+'Cebpa_ERR235722'+'_peaks.narrowPeak',\n",
    "path+'Foxa1_ERR235786'+'_peaks.narrowPeak',\n",
    "path+'Onecut1_ERR235752'+'_peaks.narrowPeak',\n",
    "'/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_VS_CONTROL_peaks.narrowPeak'\n",
    "]\n",
    "narrow_peaks_dict = {x: pr.read_bed(x) for x in files}\n",
    "# Rename\n",
    "for x in narrow_peaks_dict.keys():\n",
    "    narrow_peaks_dict[x].columns = [\n",
    "            'Chromosome',\n",
    "            'Start',\n",
    "            'End',\n",
    "            'Name',\n",
    "            'Score',\n",
    "            'Strand',\n",
    "            'FC_summit',\n",
    "            '-log10_pval',\n",
    "            '-log10_qval',\n",
    "            'Summit']\n",
    "# Get chromosome sizes\n",
    "import requests\n",
    "target_url='http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes'\n",
    "chromsizes=pd.read_csv(target_url, sep='\\t', header=None)\n",
    "chromsizes.columns=['Chromosome', 'End']\n",
    "chromsizes['Start']=[0]*chromsizes.shape[0]\n",
    "chromsizes=chromsizes.loc[:,['Chromosome', 'Start', 'End']]\n",
    "# Consensus peaks\n",
    "from pycisTopic.iterative_peak_calling import *\n",
    "# Other param\n",
    "peak_half_width=250\n",
    "path_to_blacklist='/staging/leuven/stg_00002/lcb/cbravo/Multiomics_pipeline/pycisTopic/blacklist/mm10-blacklist.v2.bed'\n",
    "# Get consensus peaks\n",
    "consensus_peaks=get_consensus_peaks(narrow_peaks_dict, peak_half_width, chromsizes=chromsizes, path_to_blacklist=path_to_blacklist)\n",
    "consensus_peaks = consensus_peaks.df\n",
    "# Write to bed\n",
    "pr.PyRanges(consensus_peaks).to_bed(path=path+'CHIP-seq_consensus_peaks_wTbx3.bed', keep=True, compression='infer', chain=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2828237f-c869-45b4-aa1c-07937b70810c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-05-22 20:32:50,549 cisTopic     INFO     Extending and merging peaks per class\n",
      "2023-05-22 20:32:52,502 cisTopic     INFO     Normalizing peak scores\n",
      "2023-05-22 20:32:52,515 cisTopic     INFO     Merging peaks\n",
      "Warning! Start and End columns now have different dtypes: int64 and int32\n",
      "2023-05-22 20:32:53,212 cisTopic     INFO     Done!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n",
      "/opt/venv/lib/python3.8/site-packages/pyranges/out.py:37: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  return pd.concat([outdf, df.get(noncanonical)], axis=1)\n"
     ]
    }
   ],
   "source": [
    "# Generate ChIP-seq consensus peaks\n",
    "import os\n",
    "import pyranges as pr\n",
    "import pandas as pd\n",
    "\n",
    "path='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/MACS/'\n",
    "files = [\n",
    "'/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_VS_CONTROL_peaks.narrowPeak'\n",
    "]\n",
    "narrow_peaks_dict = {x: pr.read_bed(x) for x in files}\n",
    "# Rename\n",
    "for x in narrow_peaks_dict.keys():\n",
    "    narrow_peaks_dict[x].columns = [\n",
    "            'Chromosome',\n",
    "            'Start',\n",
    "            'End',\n",
    "            'Name',\n",
    "            'Score',\n",
    "            'Strand',\n",
    "            'FC_summit',\n",
    "            '-log10_pval',\n",
    "            '-log10_qval',\n",
    "            'Summit']\n",
    "# Get chromosome sizes\n",
    "import requests\n",
    "target_url='http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes'\n",
    "chromsizes=pd.read_csv(target_url, sep='\\t', header=None)\n",
    "chromsizes.columns=['Chromosome', 'End']\n",
    "chromsizes['Start']=[0]*chromsizes.shape[0]\n",
    "chromsizes=chromsizes.loc[:,['Chromosome', 'Start', 'End']]\n",
    "# Consensus peaks\n",
    "from pycisTopic.iterative_peak_calling import *\n",
    "# Other param\n",
    "peak_half_width=250\n",
    "path_to_blacklist='/staging/leuven/stg_00002/lcb/cbravo/Multiomics_pipeline/pycisTopic/blacklist/mm10-blacklist.v2.bed'\n",
    "# Get consensus peaks\n",
    "consensus_peaks=get_consensus_peaks(narrow_peaks_dict, peak_half_width, chromsizes=chromsizes, path_to_blacklist=path_to_blacklist)\n",
    "consensus_peaks = consensus_peaks.df\n",
    "# Write to bed\n",
    "pr.PyRanges(consensus_peaks).to_bed(path=path+'Tbx3_CHIP-seq_consensus.bed', keep=True, compression='infer', chain=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e01455f-3408-4cc3-928f-18a0b341901f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Intersect\n",
    "module load BEDTools\n",
    "path_to_folder='/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/GEMSTAT/Region_selection_v3/regions/'\n",
    "pattern='Chip_overlap/'\n",
    "mkdir -p ${path_to_folder}${pattern}\n",
    "#file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/MACS/CHIP-seq_consensus_peaks_wTbx3.bed'\n",
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/MACS/Tbx3_CHIP-seq_consensus.bed'\n",
    "bedtools intersect -a $file -b ${path_to_folder}General_clean_v4.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}General_clean_v4_ov.bed\n",
    "bedtools intersect -a $file -b ${path_to_folder}Intermediate_clean_v4.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Intermediate_clean_v4_ov.bed\n",
    "bedtools intersect -a $file -b ${path_to_folder}Periportal_clean_v4.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Periportal_clean_v4_ov.bed\n",
    "bedtools intersect -a $file -b ${path_to_folder}Pericentral_clean_v4.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Pericentral_clean_v4_ov.bed\n",
    "\n",
    "bedtools intersect -b $file -a ${path_to_folder}General_clean_v4.bed -v -wa > ${path_to_folder}${pattern}General_clean_v4_nov.bed\n",
    "bedtools intersect -b $file -a ${path_to_folder}Intermediate_clean_v4.bed -v -wa > ${path_to_folder}${pattern}Intermediate_clean_v4_nov.bed\n",
    "bedtools intersect -b $file -a ${path_to_folder}Periportal_clean_v4.bed -v -wa > ${path_to_folder}${pattern}Periportal_clean_v4_nov.bed\n",
    "bedtools intersect -b $file -a ${path_to_folder}Pericentral_clean_v4.bed -v -wa > ${path_to_folder}${pattern}Pericentral_clean_v4_nov.bed\n",
    "\n",
    "cat ${path_to_folder}${pattern}General_clean_v4_ov.bed ${path_to_folder}${pattern}General_clean_v4_nov.bed > ${path_to_folder}${pattern}General_clean_v4.bed\n",
    "cat ${path_to_folder}${pattern}Intermediate_clean_v4_ov.bed ${path_to_folder}${pattern}Intermediate_clean_v4_nov.bed > ${path_to_folder}${pattern}Intermediate_clean_v4.bed\n",
    "cat ${path_to_folder}${pattern}Periportal_clean_v4_ov.bed ${path_to_folder}${pattern}Periportal_clean_v4_nov.bed > ${path_to_folder}${pattern}Periportal_clean_v4.bed\n",
    "cat ${path_to_folder}${pattern}Pericentral_clean_v4_ov.bed ${path_to_folder}${pattern}Pericentral_clean_v4_nov.bed > ${path_to_folder}${pattern}Pericentral_clean_v4.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34a10471-84b9-4dbb-b431-b0b1096019b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R ${path_to_folder}${pattern}General_clean_v4.bed \\\n",
    "${path_to_folder}${pattern}Pericentral_clean_v4.bed \\\n",
    "${path_to_folder}${pattern}Intermediate_clean_v4.bed \\\n",
    "${path_to_folder}${pattern}Periportal_clean_v4.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3_ChIP-seq \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 7 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal \\\n",
    "--regionsLabel General Pericentral Intermediate Periportal \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Oranges \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_ChIP_on_Zonated_regions.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acd3f4c8-f0c0-430d-8f3c-9e04e7e0a253",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R ${path_to_folder}${pattern}Pericentral_clean_v4.bed \\\n",
    "${path_to_folder}${pattern}Periportal_clean_v4.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3_ChIP-seq \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 7 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_PC_PP\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_PC_PP \\\n",
    "--regionsLabel Pericentral Periportal \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_ChIP_on_Zonated_regions_PC_PP.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3a4ea1f-7509-462b-8e96-6787b0ff322a",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R ${path_to_folder}${pattern}Pericentral_clean_v4.bed \\\n",
    "${path_to_folder}${pattern}Intermediate_clean_v4.bed \\\n",
    "${path_to_folder}${pattern}Periportal_clean_v4.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3_ChIP-seq \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 7 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_PC_PP\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_PC_PP \\\n",
    "--regionsLabel Pericentral Intermediate Periportal \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_ChIP_on_Zonated_regions_PC_PP.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa17df03-74b1-4cdc-a91d-2715aa144d6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R ${path_to_folder}${pattern}Pericentral_clean_v4_all.bed \\\n",
    "${path_to_folder}${pattern}Periportal_clean_v4_all.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3_ChIP-seq \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 7 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_PC_PP_ov\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_PC_PP_ov \\\n",
    "--regionsLabel Pericentral Periportal \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_ChIP_on_Zonated_regions_PC_PP_all_Greens.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59fa3f45-30e5-424a-88d8-00acbc19a4b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R ${path_to_folder}${pattern}Pericentral_clean_v4_ov.bed \\\n",
    "${path_to_folder}${pattern}Intermediate_clean_v4_ov.bed \\\n",
    "${path_to_folder}${pattern}Periportal_clean_v4_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3_ChIP-seq \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 7 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_PC_Int_PP_ov\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_PC_Int_PP_ov \\\n",
    "--regionsLabel Pericentral Intermediate Periportal \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_ChIP_on_Zonated_regions_PC_Int_PP_ov_Greens.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f5cf702a-c277-445f-b807-08e5a9681c58",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Intersect\n",
    "module load BEDTools\n",
    "path_to_folder='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/'\n",
    "pattern='carrot/'\n",
    "mkdir -p ${path_to_folder}${pattern}\n",
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling/consensus_regions.bed'\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Tbx3_Chip-seq_cistrome.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Tbx3_chip_consensus_ov.bed\n",
    "bedtools intersect -a /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Tbx3_Chip-seq_cistrome.bed -b $file  -v -wa > ${path_to_folder}${pattern}Tbx3_chip_consensus_nov.bed\n",
    "cat ${path_to_folder}${pattern}Tbx3_chip_consensus_ov.bed ${path_to_folder}${pattern}Tbx3_chip_consensus_nov.bed > ${path_to_folder}${pattern}Tbx3_chip_consensus.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c02cbd26-88f8-44b6-84da-a2a0111cc006",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  ${path_to_folder}${pattern}Tbx3_chip_consensus_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling_mallet/pseudobulk_bw_files/*.bw \\\n",
    "--samplesLabel BEC B_cell Fibroblast HSC Hep_PP_PC_1 Hep_PP_PC_2 Hep_PP_PC_3 Kupffer LSEC MSC T_cell VEC_PC VEC_PP cDC pDC  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 7 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_on_cell_types\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_on_cell_types \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_signal_on_cell_types.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b77f6189-c0c7-4f8e-8cf3-ce5c6c068fff",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  ${path_to_folder}${pattern}Tbx3_chip_consensus_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling/pseudobulk_bw_files/Multiome_Hep_*.bw \\\n",
    "--samplesLabel Hep_PP_PC_1 Hep_PP_PC_2 Hep_PP_PC_3 Hep_PP_PC_4 Hep_PP_PC_5  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92fd51a5-7cd8-45d8-9c5c-bca0facab638",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Intersect\n",
    "module load BEDTools\n",
    "path_to_folder='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/'\n",
    "pattern='carrot/'\n",
    "mkdir -p ${path_to_folder}${pattern}\n",
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling/consensus_regions.bed'\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_VS_CONTROL_peaks.narrowPeak -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Tbx3_narrowpeak_consensus_ov.bed\n",
    "bedtools intersect -b $file -a /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_VS_CONTROL_peaks.narrowPeak -v -wa > ${path_to_folder}${pattern}Tbx3_narrowpeak_consensus_nov.bed\n",
    "cat ${path_to_folder}${pattern}Tbx3_narrowpeak_consensus_ov.bed ${path_to_folder}${pattern}Tbx3_narrowpeak_consensus_nov.bed > ${path_to_folder}${pattern}Tbx3_anrrowpeak_consensus.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57ab1ece-0e63-4d26-87a9-830def83eda9",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  ${path_to_folder}${pattern}Tbx3_narrowpeak_consensus_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling/pseudobulk_bw_files/Multiome_Hep_*.bw \\\n",
    "--samplesLabel Hep_PP_PC_1 Hep_PP_PC_2 Hep_PP_PC_3 Hep_PP_PC_4 Hep_PP_PC_5  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO_narrowpeak\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO_narrowpeak \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO_narrowpeak.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8567687a-9621-4953-81fa-a9c081c267e0",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (518924299.py, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn [5], line 2\u001b[0;36m\u001b[0m\n\u001b[0;31m    module load BEDTools\u001b[0m\n\u001b[0m           ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "# Intersect\n",
    "module load BEDTools\n",
    "path_to_folder='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/'\n",
    "pattern='carrot/'\n",
    "mkdir -p ${path_to_folder}${pattern}\n",
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling/consensus_regions.bed'\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Tbx3_1K_consensus_ov.bed\n",
    "bedtools intersect -b $file -a /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed -v -wa > ${path_to_folder}${pattern}Tbx3_1K_consensus_nov.bed\n",
    "cat ${path_to_folder}${pattern}Tbx3_1K_consensus_ov.bed ${path_to_folder}${pattern}Tbx3_1K_consensus_nov.bed > ${path_to_folder}${pattern}Tbx3_1K_consensus.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "943f40e6-1f5c-4b91-a838-4963350e7f46",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  ${path_to_folder}${pattern}Tbx3_1K_consensus_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/consensus_peak_calling/pseudobulk_bw_files/Multiome_Hep_*.bw \\\n",
    "--samplesLabel Hep_PP_PC_1 Hep_PP_PC_2 Hep_PP_PC_3 Hep_PP_PC_4 Hep_PP_PC_5  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO_1K\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO_1K \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}Tbx3_signal_on_Hep_MO_1K.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15cb06bb-f3c0-4e1d-888e-985b740f6973",
   "metadata": {},
   "outputs": [],
   "source": [
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/scenicplus/curated_eregulons_bed_all_positive_shared/Tbx3_-.bed'\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Tbx3_Chip-seq_cistrome.bed -wa | cut -f1,2,3 | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74e0cdfa-6628-46e8-9518-e2ddbfbd98e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Overlap with tcf7l1 regions\n",
    "grep 'Tcf7l1' /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/scenicplus/final_objects/r2g_links/region_to_gene.tsv | cut -f3 | sed 's/:/\\t/g' | sed 's/-/\\t/g' > /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/scenicplus/final_objects/r2g_links/Regions_linked_to_Tcf7l1.bed\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3839fc37-e13a-4c34-ae8e-6123e6aef02b",
   "metadata": {},
   "outputs": [],
   "source": [
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/scenicplus/final_objects/r2g_links/Regions_linked_to_Tcf7l1.bed'\n",
    "perl -p -i -e 's//\\t/g' $file\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_VS_CONTROL_peaks.narrowPeak -wa > /staging/leuven/stg_00002/lcb/cbravo/Liver/Multiome/pycistopic/scenicplus/final_objects/r2g_links/Regions_linked_to_Tcf7l1_with_Tbx3_ChIPseq.bed\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "621e0498-ecd3-4870-8d1f-4343013a1d13",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Intersect\n",
    "module load BEDTools\n",
    "path_to_folder='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/'\n",
    "pattern='carrot/'\n",
    "mkdir -p ${path_to_folder}${pattern}\n",
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/MACS/CHIP-seq_consensus_peaks_wTbx3.bed'\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_ov.bed\n",
    "bedtools intersect -a /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed  -b $file  -v -wa > ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_nov.bed\n",
    "cat ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_ov.bed ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_nov.bed > ${path_to_folder}${pattern}Tbx3_1K_ChIP_consensus.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be23a985-eae6-4206-9b50-545844f2c966",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Hnf4a_ERR235763_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Cebpa_ERR235722_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Foxa1_ERR235786_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Onecut1_ERR235752_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3 Hnf4a Cebpa Foxa1 Onecut1  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K_consensus.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f1f30da-c193-405e-bae8-9a9a24126133",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Hnf4a_ERR235763_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Cebpa_ERR235722_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Foxa1_ERR235786_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Onecut1_ERR235752_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3 Hnf4a Cebpa Foxa1 Onecut1  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "845f34a2-d4ec-409b-8f39-b0e6d830f6be",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (933044688.py, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn [7], line 2\u001b[0;36m\u001b[0m\n\u001b[0;31m    module load BEDTools\u001b[0m\n\u001b[0m           ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "# Intersect\n",
    "module load BEDTools\n",
    "path_to_folder='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/'\n",
    "pattern='carrot/'\n",
    "mkdir -p ${path_to_folder}${pattern}\n",
    "file='/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/MACS/Tbx3_CHIP-seq_consensus.bed'\n",
    "bedtools intersect -a $file -b /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed -wa | cut -f1,2,3 > ${path_to_folder}${pattern}Tbx3_1K_consensusTbx3_ChIP_ov.bed\n",
    "bedtools intersect -a /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/MACS0.01/Top1K_Tbx3_peaks.bed  -b $file  -v -wa > ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_Tbx3nov.bed\n",
    "cat ${path_to_folder}${pattern}Tbx3_1K_consensusTbx3_ChIP_ov.bed ${path_to_folder}${pattern}Tbx3_1K_consensus_ChIP_Tbx3nov.bed > ${path_to_folder}${pattern}Tbx3_1K_ChIP_consensusTbx3.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "210551fb-a0bd-4b6c-a1a1-eb886cda52da",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  ${path_to_folder}${pattern}Tbx3_1K_consensusTbx3_ChIP_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Hnf4a_ERR235763_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Cebpa_ERR235722_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Foxa1_ERR235786_q4_sorted.bw \\\n",
    "/staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/bam/Onecut1_ERR235752_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3 Hnf4a Cebpa Foxa1 Onecut1  \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K_wTbx3_peals.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63fc59c9-b4dc-4c1e-859d-fec0d9c5b6f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load Python/3.9.5-GCCcore-10.3.0\n",
    "cd ${path_to_folder}${pattern}\n",
    "computeMatrix scale-regions \\\n",
    "-R  ${path_to_folder}${pattern}Tbx3_1K_consensusTbx3_ChIP_ov.bed \\\n",
    "-S /staging/leuven/stg_00002/lcb/cbravo/Liver/Liver_CHIP/Inhouse_Tbx3/bam/MLV__0a3a08__ChIPseq_mouse_hepatocytes_anti-TBX3_Bethyl_rabbit_S29_q4_sorted.bw \\\n",
    "--samplesLabel Tbx3 \\\n",
    "--missingDataAsZero \\\n",
    "--sortRegions descend \\\n",
    "--sortUsing mean \\\n",
    "-p 20 \\\n",
    "-o ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K\n",
    "\n",
    "plotHeatmap \\\n",
    "--matrixFile ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K \\\n",
    "--regionsLabel Tbx3_ChIP \\\n",
    "--sortUsing mean \\\n",
    "--colorMap Greens \\\n",
    "--dpi 800 \\\n",
    "--outFileName ${path_to_folder}${pattern}ChIPseq_signal_on_Tbx3_1K_wTbx3_peals.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48ea2003-eed7-483a-a225-367400bf0540",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "DEV IMAGE - SCENIC+ latest (use at own risk)",
   "language": "python",
   "name": "pyscenic_plus_dev"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
